<!doctype html><html lang=en dir=auto><head><meta charset=utf-8><meta http-equiv=x-ua-compatible content="IE=edge"><meta name=viewport content="width=device-width,initial-scale=1,shrink-to-fit=no"><meta name=robots content="index, follow"><title>Transcription factor binding site的基礎概念 | Biology and computer science</title><meta name=keywords content="PMW,TF,regulation"><meta name=description content="描述TF binding site與簡單預測的方法"><meta name=author content="Me"><link rel=canonical href=https://canonical.url/to/page><link crossorigin=anonymous href=/assets/css/stylesheet.bccfefac377bc340f06c260aed1bddf49a4354816d7c570d6aac75a997986c95.css integrity="sha256-vM/vrDd7w0DwbCYK7Rvd9JpDVIFtfFcNaqx1qZeYbJU=" rel="preload stylesheet" as=style><script defer crossorigin=anonymous src=/assets/js/highlight.f413e19d0714851f6474e7ee9632408e58ac146fbdbe62747134bea2fa3415e0.js integrity="sha256-9BPhnQcUhR9kdOfuljJAjlisFG+9vmJ0cTS+ovo0FeA=" onload=hljs.initHighlightingOnLoad()></script>
<link rel=icon href=https://xavierzheng.github.io/%3Clink%20/%20abs%20url%3E><link rel=icon type=image/png sizes=16x16 href=https://xavierzheng.github.io/%3Clink%20/%20abs%20url%3E><link rel=icon type=image/png sizes=32x32 href=https://xavierzheng.github.io/%3Clink%20/%20abs%20url%3E><link rel=apple-touch-icon href=https://xavierzheng.github.io/%3Clink%20/%20abs%20url%3E><link rel=mask-icon href=https://xavierzheng.github.io/%3Clink%20/%20abs%20url%3E><meta name=theme-color content="#2e2e33"><meta name=msapplication-TileColor content="#2e2e33"><noscript><style>#theme-toggle,.top-link{display:none}</style><style>@media(prefers-color-scheme:dark){:root{--theme:rgb(29, 30, 32);--entry:rgb(46, 46, 51);--primary:rgb(218, 218, 219);--secondary:rgb(155, 156, 157);--tertiary:rgb(65, 66, 68);--content:rgb(196, 196, 197);--hljs-bg:rgb(46, 46, 51);--code-bg:rgb(55, 56, 62);--border:rgb(51, 51, 51)}.list{background:var(--theme)}.list:not(.dark)::-webkit-scrollbar-track{background:0 0}.list:not(.dark)::-webkit-scrollbar-thumb{border-color:var(--theme)}}</style></noscript><meta property="og:title" content="Transcription factor binding site的基礎概念"><meta property="og:description" content="描述TF binding site與簡單預測的方法"><meta property="og:type" content="article"><meta property="og:url" content="https://xavierzheng.github.io/posts/tfbindingsite/"><meta property="og:image" content="https://xavierzheng.github.io/%3Cimage%20path/url%3E"><meta property="article:section" content="posts"><meta property="article:published_time" content="2022-12-30T12:13:35+08:00"><meta property="article:modified_time" content="2022-12-30T12:13:35+08:00"><meta property="og:site_name" content="Biology and computer science"><meta name=twitter:card content="summary_large_image"><meta name=twitter:image content="https://xavierzheng.github.io/%3Cimage%20path/url%3E"><meta name=twitter:title content="Transcription factor binding site的基礎概念"><meta name=twitter:description content="描述TF binding site與簡單預測的方法"><script type=application/ld+json>{"@context":"https://schema.org","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","position":2,"name":"Posts","item":"https://xavierzheng.github.io/posts/"},{"@type":"ListItem","position":3,"name":"Transcription factor binding site的基礎概念","item":"https://xavierzheng.github.io/posts/tfbindingsite/"}]}</script><script type=application/ld+json>{"@context":"https://schema.org","@type":"BlogPosting","headline":"Transcription factor binding site的基礎概念","name":"Transcription factor binding site的基礎概念","description":"描述TF binding site與簡單預測的方法","keywords":["PMW","TF","regulation"],"articleBody":"Transcription factor binding site Transcription factor binding site是實驗能證明的區域，這段序列能與transcription factor 結合，進而開啟或關閉基因表現。本文沒有要強調Transcription factor binding site如何尋找或進行實驗驗證，僅介紹如何描述這段transcription factor binding site。\n字串描述 最簡單的描述方法是CCATATATAG 這樣的單純字串描述，如下圖所示 （https://jaspar.genereg.net/sites/MA0001.1/）\n這種顯示方式最直觀，也最好利用，大家可以想像只要使用Ctrl + F或grep等方式進行搜尋。\n正規表示法描述 有些位置如果有超過一種以上的DNA出現，這時候單純用ATCGU表現就不太方便了。目前IUPAC有訂立一系列標準用字去描述這些可能性，請參考 https://en.m.wikipedia.org/wiki/Nucleic_acid_notation ，就像R代表purine，包含AG。而Y代表Pyrimidine，包含CT等等。\nPosition weight matrix (PWM) 其實，transcription factor binding site的專一性很少有達到需每一個base都正確才有效能。通常transcription factor binding site是由一堆可能性堆疊而成，它描述了每個位置出現的ATCG頻率。以下圖左邊為例，直的column 代表每個位置，而橫的row代表ATCG，所以矩陣裡面的數值就是每個位置 （column）裡，ATCG出現的頻率。以第一個位置而言，看起來C出現的頻率最高，但Q或G仍有出現之可能。\nPMW DNAlogo Ref. https://jaspar.genereg.net/matrix/MA0001.1/?revcomp=1\n通常這種描述太數學了，如果想更視覺化一點，我們會把這些序列轉成DNAlogo的型態來描述，如上圖右側所示，因為第一個位置C出現的機率遠大約於其他核酸，所以就會看到一根很大的C。\n好，那position weighted matrix看起來最實際，但該怎樣用它來搜尋全基因體，或是promoter region呢？\nMEME package FIMO找序列 簡單暴力解，MEME package 裡面的FIMO專做這件事情 (https://meme-suite.org/meme/tools/fimo)，只要有PWM，並改成MEME家的格式，餵給FIMO，再給一段序列，FIMO就能幫忙尋找這段PWM是否存在給定序列裡了。\n這樣有簡單的輸出結果可以參考 https://meme-suite.org/meme/doc/examples/fimo_example_output_files/fimo.html 。MEME package 最貼心的是，他會在最下面列出現在使用的指令是什麼，你可以使用這指令在自己的電腦上重現這些結果\n# Command line: Fimo –oc fimo_example_output_files –parse-genomic-coord some_vertebrates.meme mm9_tss_500bp_sampled_1000.fna # Settings: Output_directory = fimo_example_output_files\tMEME file name = some_vertebrates.meme\tsequence file name = mm9_tss_500bp_sampled_1000.fna Background file name = --nrdb--\talphabet = DNA\tmax stored scores = 100000 Allow clobber = true\tcompute q-values = true\tparse genomic coord. = true Text only = false\tscan both strands = true\tmax strand = false Threshold type = p-value\toutput theshold = 0.0001\tpseudocount = 0.1 Alpha = 1\tverbosity = 2\tMEME package MEME建立PWM 那如何建立PMW呢？最棒的方法是用multiple sequence alignment，真實的看到了某堆序列裡，真的有一段conserved sequence (高度保留序列)，我們再拿這個高度保留序列製成PMW。但這挑戰性極高，DNA multiple sequence alignment 沒那麼好做，所以簡單做法是透過生物實驗，像是RNA sequence找出coexpression network，或是用kmean把具有一致expression patterns 的基因們打撈出來，甚至做chip seq，直接測量蛋白質與DNA的結合。最後再拿這些資料給MEME，讓他幫忙找在這些序列裡是不是真的藏有一些短片段。\n注意：MEME定義的PMW motif有兩大前提假設，第一是gapless，也就是禁止出現gap。第二，長度要一致。也就是說MEME如果發現兩個長度不一致的motif，它會直接定義成兩個motif\n實際範例：https://meme-suite.org/meme/doc/examples/meme_example_output_files/meme.html?man_type=web\nmeme lex0.fna -oc meme_example_output_files -dna -mod zoops -nmotifs 3 -revcomp -mpi 背後原理 MEME背後是進行EM (Expectation-Maximization Algorithm)。實際範例可以參考 RNA motif discovery: a computational overview | Biology Direct | Full Text (biomedcentral.com)。下圖A是一個典型的PMW，而圖B是實際應用。X代表序列，而Z代表有沒有搜尋到motif，0代表沒有，1代表有，且紅色的序列就代表了motif。由於第一個位置與許A與G的機率很高，所以可以看到A、G都能當成第一個序列\n","wordCount":"189","inLanguage":"en","image":"https://xavierzheng.github.io/%3Cimage%20path/url%3E","datePublished":"2022-12-30T12:13:35+08:00","dateModified":"2022-12-30T12:13:35+08:00","author":{"@type":"Person","name":"Me"},"mainEntityOfPage":{"@type":"WebPage","@id":"https://xavierzheng.github.io/posts/tfbindingsite/"},"publisher":{"@type":"Organization","name":"Biology and computer science","logo":{"@type":"ImageObject","url":"https://xavierzheng.github.io/%3Clink%20/%20abs%20url%3E"}}}</script></head><body id=top><script>localStorage.getItem("pref-theme")==="dark"?document.body.classList.add("dark"):localStorage.getItem("pref-theme")==="light"?document.body.classList.remove("dark"):window.matchMedia("(prefers-color-scheme: dark)").matches&&document.body.classList.add("dark")</script><header class=header><nav class=nav><div class=logo><a href=https://xavierzheng.github.io accesskey=h title="Biology and computer science (Alt + H)"><img src=https://xavierzheng.github.io/apple-touch-icon.png alt aria-label=logo height=35>Biology and computer science</a><div class=logo-switches><button id=theme-toggle accesskey=t title="(Alt + T)"><svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24" fill="none" stroke="currentcolor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><path d="M21 12.79A9 9 0 1111.21 3 7 7 0 0021 12.79z"/></svg><svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24" fill="none" stroke="currentcolor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><circle cx="12" cy="12" r="5"/><line x1="12" y1="1" x2="12" y2="3"/><line x1="12" y1="21" x2="12" y2="23"/><line x1="4.22" y1="4.22" x2="5.64" y2="5.64"/><line x1="18.36" y1="18.36" x2="19.78" y2="19.78"/><line x1="1" y1="12" x2="3" y2="12"/><line x1="21" y1="12" x2="23" y2="12"/><line x1="4.22" y1="19.78" x2="5.64" y2="18.36"/><line x1="18.36" y1="5.64" x2="19.78" y2="4.22"/></svg></button></div></div><ul id=menu><li><a href=https://xavierzheng.github.io/search title="search (Alt + /)" accesskey=/><span>search</span></a></li><li><a href=https://xavierzheng.github.io/tags/ title=tags><span>tags</span></a></li><li><a href=https://xavierzheng.github.io/archives/ title=archives><span>archives</span></a></li></ul></nav></header><main class=main><article class=post-single><header class=post-header><div class=breadcrumbs><a href=https://xavierzheng.github.io>Home</a>&nbsp;»&nbsp;<a href=https://xavierzheng.github.io/posts/>Posts</a></div><h1 class=post-title>Transcription factor binding site的基礎概念</h1><div class=post-description>描述TF binding site與簡單預測的方法</div><div class=post-meta><span title='2022-12-30 12:13:35 +0800 CST'>December 30, 2022</span>&nbsp;·&nbsp;Me</div></header><div class=toc><details open><summary accesskey=c title="(Alt + C)"><span class=details>Table of Contents</span></summary><div class=inner><nav id=TableOfContents><ul><li><a href=#字串描述>字串描述</a></li><li><a href=#正規表示法描述>正規表示法描述</a></li><li><a href=#position-weight-matrix-pwm>Position weight matrix (PWM)</a></li><li><a href=#meme-package-fimo找序列>MEME package FIMO找序列</a></li><li><a href=#meme-package-meme建立pwm>MEME package MEME建立PWM</a></li><li><a href=#背後原理>背後原理</a></li></ul></nav></div></details></div><div class=post-content><h1 id=transcription-factor-binding-site>Transcription factor binding site<a hidden class=anchor aria-hidden=true href=#transcription-factor-binding-site>#</a></h1><p>Transcription factor binding site是實驗能證明的區域，這段序列能與transcription factor 結合，進而開啟或關閉基因表現。本文沒有要強調Transcription factor binding site如何尋找或進行實驗驗證，僅介紹如何描述這段transcription factor binding site。</p><h2 id=字串描述>字串描述<a hidden class=anchor aria-hidden=true href=#字串描述>#</a></h2><p>最簡單的描述方法是CCATATATAG 這樣的單純字串描述，如下圖所示 （https://jaspar.genereg.net/sites/MA0001.1/）</p><p><img loading=lazy src=/TFBindingSite.assets/clip_image002.jpg alt=String></p><p>這種顯示方式最直觀，也最好利用，大家可以想像只要使用Ctrl + F或grep等方式進行搜尋。</p><h2 id=正規表示法描述>正規表示法描述<a hidden class=anchor aria-hidden=true href=#正規表示法描述>#</a></h2><p>有些位置如果有超過一種以上的DNA出現，這時候單純用ATCGU表現就不太方便了。目前IUPAC有訂立一系列標準用字去描述這些可能性，請參考 <a href=https://en.m.wikipedia.org/wiki/Nucleic_acid_notation>https://en.m.wikipedia.org/wiki/Nucleic_acid_notation</a> ，就像R代表purine，包含AG。而Y代表Pyrimidine，包含CT等等。</p><h2 id=position-weight-matrix-pwm>Position weight matrix (PWM)<a hidden class=anchor aria-hidden=true href=#position-weight-matrix-pwm>#</a></h2><p>其實，transcription factor binding site的專一性很少有達到需每一個base都正確才有效能。通常transcription factor binding site是由一堆可能性堆疊而成，它描述了每個位置出現的ATCG頻率。以下圖左邊為例，直的column 代表每個位置，而橫的row代表ATCG，所以矩陣裡面的數值就是每個位置 （column）裡，ATCG出現的頻率。以第一個位置而言，看起來C出現的頻率最高，但Q或G仍有出現之可能。</p><table><thead><tr><th>PMW</th><th>DNAlogo</th></tr></thead><tbody><tr><td><img loading=lazy src=/TFBindingSite.assets/clip_image004.jpg alt="PMW example"></td><td><img loading=lazy src=/TFBindingSite.assets/clip_image003.jpg alt=DNAlogo></td></tr></tbody></table><p>Ref. <a href="https://jaspar.genereg.net/matrix/MA0001.1/?revcomp=1">https://jaspar.genereg.net/matrix/MA0001.1/?revcomp=1</a></p><p>通常這種描述太數學了，如果想更視覺化一點，我們會把這些序列轉成DNAlogo的型態來描述，如上圖右側所示，因為第一個位置C出現的機率遠大約於其他核酸，所以就會看到一根很大的C。</p><p>好，那position weighted matrix看起來最實際，但該怎樣用它來搜尋全基因體，或是promoter region呢？</p><h2 id=meme-package-fimo找序列>MEME package FIMO找序列<a hidden class=anchor aria-hidden=true href=#meme-package-fimo找序列>#</a></h2><p>簡單暴力解，MEME package 裡面的FIMO專做這件事情 (<a href=https://meme-suite.org/meme/tools/fimo>https://meme-suite.org/meme/tools/fimo</a>)，只要有PWM，並改成MEME家的格式，餵給FIMO，再給一段序列，FIMO就能幫忙尋找這段PWM是否存在給定序列裡了。</p><p>這樣有簡單的輸出結果可以參考 <a href=https://meme-suite.org/meme/doc/examples/fimo_example_output_files/fimo.html>https://meme-suite.org/meme/doc/examples/fimo_example_output_files/fimo.html</a> 。MEME package 最貼心的是，他會在最下面列出現在使用的指令是什麼，你可以使用這指令在自己的電腦上重現這些結果</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span><span style=color:#75715e># Command line:</span>
</span></span><span style=display:flex><span>Fimo –oc fimo_example_output_files –parse-genomic-coord some_vertebrates.meme mm9_tss_500bp_sampled_1000.fna
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># Settings:</span>
</span></span><span style=display:flex><span>Output_directory <span style=color:#f92672>=</span> fimo_example_output_files	
</span></span><span style=display:flex><span>MEME file name <span style=color:#f92672>=</span> some_vertebrates.meme	sequence file name <span style=color:#f92672>=</span> mm9_tss_500bp_sampled_1000.fna
</span></span><span style=display:flex><span>Background file name <span style=color:#f92672>=</span> --nrdb--	
</span></span><span style=display:flex><span>alphabet <span style=color:#f92672>=</span> DNA	
</span></span><span style=display:flex><span>max stored scores <span style=color:#f92672>=</span> <span style=color:#ae81ff>100000</span>
</span></span><span style=display:flex><span>Allow clobber <span style=color:#f92672>=</span> true	
</span></span><span style=display:flex><span>compute q-values <span style=color:#f92672>=</span> true	parse genomic coord. <span style=color:#f92672>=</span> true
</span></span><span style=display:flex><span>Text only <span style=color:#f92672>=</span> false	
</span></span><span style=display:flex><span>scan both strands <span style=color:#f92672>=</span> true	max strand <span style=color:#f92672>=</span> false
</span></span><span style=display:flex><span>Threshold type <span style=color:#f92672>=</span> p-value	
</span></span><span style=display:flex><span>output theshold <span style=color:#f92672>=</span> 0.0001	
</span></span><span style=display:flex><span>pseudocount <span style=color:#f92672>=</span> 0.1
</span></span><span style=display:flex><span>Alpha <span style=color:#f92672>=</span> 1	verbosity <span style=color:#f92672>=</span> 2	
</span></span></code></pre></div><h2 id=meme-package-meme建立pwm>MEME package MEME建立PWM<a hidden class=anchor aria-hidden=true href=#meme-package-meme建立pwm>#</a></h2><p>那如何建立PMW呢？最棒的方法是用multiple sequence alignment，真實的看到了某堆序列裡，真的有一段conserved sequence (高度保留序列)，我們再拿這個高度保留序列製成PMW。但這挑戰性極高，DNA multiple sequence alignment 沒那麼好做，所以簡單做法是透過生物實驗，像是RNA sequence找出coexpression network，或是用kmean把具有一致expression patterns 的基因們打撈出來，甚至做chip seq，直接測量蛋白質與DNA的結合。最後再拿這些資料給MEME，讓他幫忙找在這些序列裡是不是真的藏有一些短片段。</p><p>注意：MEME定義的PMW motif有兩大前提假設，<strong>第一是gapless</strong>，也就是禁止出現gap。<strong>第二，長度要一致</strong>。也就是說MEME如果發現兩個長度不一致的motif，它會直接定義成兩個motif</p><p>實際範例：https://meme-suite.org/meme/doc/examples/meme_example_output_files/meme.html?man_type=web</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span>meme lex0.fna -oc meme_example_output_files -dna -mod zoops -nmotifs <span style=color:#ae81ff>3</span> -revcomp -mpi
</span></span></code></pre></div><h2 id=背後原理>背後原理<a hidden class=anchor aria-hidden=true href=#背後原理>#</a></h2><p>MEME背後是進行EM (Expectation-Maximization Algorithm)。實際範例可以參考 <a href=https://biologydirect.biomedcentral.com/articles/10.1186/s13062-015-0090-5>RNA motif discovery: a computational overview | Biology Direct | Full Text (biomedcentral.com)</a>。下圖A是一個典型的PMW，而圖B是實際應用。X代表序列，而Z代表有沒有搜尋到motif，0代表沒有，1代表有，且紅色的序列就代表了motif。由於第一個位置與許A與G的機率很高，所以可以看到A、G都能當成第一個序列</p><p><img loading=lazy src=/TFBindingSite.assets/clip_image002.png alt="EM algorism"></p></div><footer class=post-footer><ul class=post-tags><li><a href=https://xavierzheng.github.io/tags/pmw/>PMW</a></li><li><a href=https://xavierzheng.github.io/tags/tf/>TF</a></li><li><a href=https://xavierzheng.github.io/tags/regulation/>regulation</a></li></ul><nav class=paginav><a class=next href=https://xavierzheng.github.io/posts/%E7%94%A8hugo%E5%92%8Cgithub%E8%A3%BD%E4%BD%9C%E9%9D%9C%E6%85%8B%E7%B6%B2%E7%AB%99/><span class=title>Next »</span><br><span>用HUGO和github製作靜態網站</span></a></nav></footer></article></main><footer class=footer><span>&copy; 2022 <a href=https://xavierzheng.github.io>Biology and computer science</a></span>
<span>Powered by
<a href=https://gohugo.io/ rel="noopener noreferrer" target=_blank>Hugo</a> &
        <a href=https://github.com/adityatelange/hugo-PaperMod/ rel=noopener target=_blank>PaperMod</a></span></footer><a href=#top aria-label="go to top" title="Go to Top (Alt + G)" class=top-link id=top-link accesskey=g><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 6" fill="currentcolor"><path d="M12 6H0l6-6z"/></svg></a><script>let menu=document.getElementById("menu");menu&&(menu.scrollLeft=localStorage.getItem("menu-scroll-position"),menu.onscroll=function(){localStorage.setItem("menu-scroll-position",menu.scrollLeft)}),document.querySelectorAll('a[href^="#"]').forEach(e=>{e.addEventListener("click",function(e){e.preventDefault();var t=this.getAttribute("href").substr(1);window.matchMedia("(prefers-reduced-motion: reduce)").matches?document.querySelector(`[id='${decodeURIComponent(t)}']`).scrollIntoView():document.querySelector(`[id='${decodeURIComponent(t)}']`).scrollIntoView({behavior:"smooth"}),t==="top"?history.replaceState(null,null," "):history.pushState(null,null,`#${t}`)})})</script><script>var mybutton=document.getElementById("top-link");window.onscroll=function(){document.body.scrollTop>800||document.documentElement.scrollTop>800?(mybutton.style.visibility="visible",mybutton.style.opacity="1"):(mybutton.style.visibility="hidden",mybutton.style.opacity="0")}</script><script>document.getElementById("theme-toggle").addEventListener("click",()=>{document.body.className.includes("dark")?(document.body.classList.remove("dark"),localStorage.setItem("pref-theme","light")):(document.body.classList.add("dark"),localStorage.setItem("pref-theme","dark"))})</script><script>document.querySelectorAll("pre > code").forEach(e=>{const n=e.parentNode.parentNode,t=document.createElement("button");t.classList.add("copy-code"),t.innerHTML="copy";function s(){t.innerHTML="copied!",setTimeout(()=>{t.innerHTML="copy"},2e3)}t.addEventListener("click",t=>{if("clipboard"in navigator){navigator.clipboard.writeText(e.textContent),s();return}const n=document.createRange();n.selectNodeContents(e);const o=window.getSelection();o.removeAllRanges(),o.addRange(n);try{document.execCommand("copy"),s()}catch{}o.removeRange(n)}),n.classList.contains("highlight")?n.appendChild(t):n.parentNode.firstChild==n||(e.parentNode.parentNode.parentNode.parentNode.parentNode.nodeName=="TABLE"?e.parentNode.parentNode.parentNode.parentNode.parentNode.appendChild(t):e.parentNode.appendChild(t))})</script></body></html>