<!doctype html><html lang=en dir=auto><head><meta charset=utf-8><meta http-equiv=x-ua-compatible content="IE=edge"><meta name=viewport content="width=device-width,initial-scale=1,shrink-to-fit=no"><meta name=robots content="index, follow"><title>A multi-omics integrative network map of maize | Biology and computer science</title><meta name=keywords content="network,paper"><meta name=description content="如何拆解multi-omics network的好概念，與文章經典格式"><meta name=author content="Me"><link rel=canonical href=https://canonical.url/to/page><link crossorigin=anonymous href=/assets/css/stylesheet.bccfefac377bc340f06c260aed1bddf49a4354816d7c570d6aac75a997986c95.css integrity="sha256-vM/vrDd7w0DwbCYK7Rvd9JpDVIFtfFcNaqx1qZeYbJU=" rel="preload stylesheet" as=style><script defer crossorigin=anonymous src=/assets/js/highlight.f413e19d0714851f6474e7ee9632408e58ac146fbdbe62747134bea2fa3415e0.js integrity="sha256-9BPhnQcUhR9kdOfuljJAjlisFG+9vmJ0cTS+ovo0FeA=" onload=hljs.initHighlightingOnLoad()></script>
<link rel=icon href=https://xavierzheng.github.io/%3Clink%20/%20abs%20url%3E><link rel=icon type=image/png sizes=16x16 href=https://xavierzheng.github.io/%3Clink%20/%20abs%20url%3E><link rel=icon type=image/png sizes=32x32 href=https://xavierzheng.github.io/%3Clink%20/%20abs%20url%3E><link rel=apple-touch-icon href=https://xavierzheng.github.io/%3Clink%20/%20abs%20url%3E><link rel=mask-icon href=https://xavierzheng.github.io/%3Clink%20/%20abs%20url%3E><meta name=theme-color content="#2e2e33"><meta name=msapplication-TileColor content="#2e2e33"><noscript><style>#theme-toggle,.top-link{display:none}</style><style>@media(prefers-color-scheme:dark){:root{--theme:rgb(29, 30, 32);--entry:rgb(46, 46, 51);--primary:rgb(218, 218, 219);--secondary:rgb(155, 156, 157);--tertiary:rgb(65, 66, 68);--content:rgb(196, 196, 197);--hljs-bg:rgb(46, 46, 51);--code-bg:rgb(55, 56, 62);--border:rgb(51, 51, 51)}.list{background:var(--theme)}.list:not(.dark)::-webkit-scrollbar-track{background:0 0}.list:not(.dark)::-webkit-scrollbar-thumb{border-color:var(--theme)}}</style></noscript><meta property="og:title" content="A multi-omics integrative network map of maize"><meta property="og:description" content="如何拆解multi-omics network的好概念，與文章經典格式"><meta property="og:type" content="article"><meta property="og:url" content="https://xavierzheng.github.io/posts/maize-network/"><meta property="og:image" content="https://xavierzheng.github.io/%3Cimage%20path/url%3E"><meta property="article:section" content="posts"><meta property="article:published_time" content="2022-12-31T08:00:32+08:00"><meta property="article:modified_time" content="2022-12-31T08:00:32+08:00"><meta property="og:site_name" content="Biology and computer science"><meta name=twitter:card content="summary_large_image"><meta name=twitter:image content="https://xavierzheng.github.io/%3Cimage%20path/url%3E"><meta name=twitter:title content="A multi-omics integrative network map of maize"><meta name=twitter:description content="如何拆解multi-omics network的好概念，與文章經典格式"><script type=application/ld+json>{"@context":"https://schema.org","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","position":2,"name":"Posts","item":"https://xavierzheng.github.io/posts/"},{"@type":"ListItem","position":3,"name":"A multi-omics integrative network map of maize","item":"https://xavierzheng.github.io/posts/maize-network/"}]}</script><script type=application/ld+json>{"@context":"https://schema.org","@type":"BlogPosting","headline":"A multi-omics integrative network map of maize","name":"A multi-omics integrative network map of maize","description":"如何拆解multi-omics network的好概念，與文章經典格式","keywords":["network","paper"],"articleBody":"念書筆記 ref: A multi-omics integrative network map of maize | Nature Genetics https://doi.org/10.1038/s41588-022-01262-1\n摘要 Evolution of gene function is often preceded by whole-genome duplication (WGD), tandem duplication (tandem), proximal duplication (proximal) or duplications mediated by transpositions (transposed) and duplications of dispersed origin (dispersed) (Fig. 2a)20,21. Network modules group genes that potentially participate in the same process. The shortest distance (SD; Methods) is a metric that represents the shortest path between a pair of nodes within a network. When applied to duplicated genes, a large SD suggests that the two genes are functionally different. All duplicated genes had a significantly higher probability of belonging to the same module (Fig. 2b) and had a lower SD (Fig. 2c) than non-paralogous genes. Tandem and proximal genes were most likely to be enriched in the same network modules, whereas transposed and dispersed genes were least likely. Gene pairs resulting from WGD, tandem and proximal duplications had the lowest SD values, indicating that they are the least functionally divergent, whereas those derived from transposed and dispersed duplications had the highest SD values, suggesting that they are the most functionally divergent (Fig. 2b,c).\n可以把基因功能的演化分成\nWhole genome duplication\nTandem duplication\nProximal duplication\nDuplication mediated by transposition\nDispersed\n在網路結構裡，可以計算兩點間最短距離（short distance)來呈現基因之間的關係。如果最短距離較大，代表兩個基因功能比較不像了。\n本研究顯示多數duplicate gene還是在同module裏面，而且最短距離也比non paralougs還要短\nTandem與proximal通常都在同一個module裡，而transpose比較不會在同一個module\n以最短距離來比較，whole genome duplication、tandem、proximal通常有較短距離，但transpose、 disperse比較長，代表他們基因功能開始不像了\nTo further explore the network divergence of duplicated gene pairs from an evolutionary perspective, we introduced two indices, the Simpson index (SAB) and the divergence index (DAB). The SAB evaluates the similarity between two nodes, while the DAB measures the network divergence of two genes, and these indices are collectively a good indicator of functional divergence (Fig. 2d). Based on SAB and DAB values, we classified all duplicated genes into five divergence patterns (Methods), with a clear gradual decrease seen in the extent of network similarity between the two genes as differentiation increased from type I to type V (Fig. 2d). These five patterns potentially represent various types of functional evolution: conservation (type I), subfunctionalization (type II), neofunctionalization (types III and IV) and nonfunctionalization/specialization (type V)22.\n利用Simpson index (SAB)與divergence index (DAB)可以把所有的duplicate gene拆分成五種。當然這是有reference 參考的分法\nconservation (type I),\nsubfunctionalization (type II),\nneofunctionalization (types III and IV)\nnonfunctionalization/specialization (type V)\nNotably, overall the enrichment ratios gradually decreased from type I to type V compared to background, indicating that duplicated genes share a stronger network similarity than would be expected by chance (Fig. 2e). Different types of duplications, such as WGD and tandem, showed the same trend (Fig. 2f and Extended Data Fig. 5). Moreover, the proportions of different types of network divergence varied across genetic layers (Fig. 2g and Extended Data Fig. 5). Interestingly, we observed a dynamic rewiring of networks after gene duplication events from the coexpression to the cotranslation to the interactome level. Our results indicate that approximately 41% of the duplicates were consistent across different omics layer, while approximately 35% were associated with network divergence and approximately 24% were associated with network convergence (Supplementary Fig. 9). For example, the duplicated genes Zm00001d049815 and Zm00001d031611 displayed a progressive network rewiring from the coexpression to the cotranslation to the interactome level (Fig. 2h). Importantly, the proportions of duplicated genes originating at different times in the past varied dramatically from type I to V, suggesting that nonfunctionalization, neofunctionalization and specialization are more likely to be associated with older duplications, whereas functional conservation is more likely for the most recent duplications (Fig. 2i). This relationship of network divergence to duplication age became clearer from the coexpression to the cotranslation to the interactome. Together, these results indicate that our integrative network map encompasses potentially evolutionary information relevant to the dissection of gene function in an evolutionary perspective.\n這種奇異的分類方法之後要接一段敘述，分析利用這種分類方法的優勢及可以看出什麼特徵，再舉個實例證實。當然這篇的優勢在於multiple omits，所有需要將這優勢資料貫穿在分析之中。 Variable network divergence of maize subgenomes Maize is an allotetraploid species with distinct ancient subgenomes, designated here as maize1 and maize2, which exhibit asymmetric divergence in both gene content and expression levels23. Although maize1, the dominant subgenome, contains more orthologs and highly expressed genes than maize2, past studies did not find regulating bias, probably because they only assessed this question at a single level23,24. Therefore, we dissected the regulating bias of the two subgenomes in more detail by classifying all subgenome genes into two groups: maize1 genes with retained maize2 orthologs and maize2 genes with retained maize1 orthologs. We then compared the degree (a network attribute; the number of edges of a node) of genes from each group at different omics levels. To this end, we calculated the variation in the degree of genes from the two different groups. We detected no significant differences between the subgenomes at the coexpression level. The difference increased but was still not significant at the cotranslation level. However, we observed a significant difference at the interactome level between the two subgenomes, as reflected by the degree values (Fig. 3a), which suggests the functional importance of genes and nodes25. Additionally, we quantified the overall difference of the degree of genes in a sliding window (100 genes) in the ancient maize subgenomes derived from the Sorghum genome as previously published23. We detected more dominant bins for the maize1 subgenome from the coexpression to the cotranslation to the interactome level, suggesting that differences in network connectivity of the subgenomes vary across the three network levels (Fig. 3b and Supplementary Figs. 10 and 11)\n植物資料的優勢在於通常有subgenome演化的問題，所有如果能以degree去區分不同subgenome差異，也會是相當有意義。 Functional networks of important genes The integrative network map can reconstruct the functional connections of well-known genes and uncover new regulating genes with similar functions (Fig. 4). We explored a subnetwork involving three key tillering genes, teosinte branched 1 (Tb1, Zm00001d033673)26, grassy tillers 1 (Gt1, Zm00001d028129)27 and tassels replace upper ears 1 (Tru1, Zm00001d042111)28,29,30,31,32,33,34. Loss-of-function mutations in any of these genes lead to the formation of more tillers, a characteristic of the maize ancestor teosinte. Tb1, Gt1 and Tru1 were all part of a coexpression network with ZmALOG1 (Zm00001d003057) and ZmALOG2 (Zm00001d032696), two genes of unknown function belonging to the Arabidopsis LSH1 and Oryza G1 (ALOG) transcription factor family (Fig. 4a). Notably, the loss-of-function mutations in ZmALOG1 and ZmALOG2 obtained by CRISPR- or CRISPR*–*Cas9-mediated gene editing led to higher tillering, similar to the tb1, gt1 and tru1 mutants (Fig. 4b and Extended Data Fig. 6a). Moreover, yeast two-hybrid assays indicated that ZmALOG1 and ZmALOG2 can interact with TB1 (Fig. 4c). These results demonstrate that the network edges between known and unknown genes in subnetworks have potential biological meaning. Therefore, our maize network map can be used to generate hypotheses about the functions of genes of interest, as well as their putative network pathways.\n建立network之後，除了串聯已知基因之外，也可以探討未知基因的功能。所以可以靠突變是否有類似功能之類的資訊，以及yeast two hybrid等方法實際證實未知基因的功能。 Predicting novel functional genes for FT FT is an important agronomic trait involved in the adaptation of maize to a wide range of climates worldwide40,41,42. Although many key FT genes in maize have been cloned (Supplementary Table 11), much remains to be learned about the genetic control and molecular mechanisms underlying this trait. To explore the potential regulating networks, we constructed a prediction model based on the integrative map and a machine learning method using known FT genes (functionally cloned FT genes in maize and maize homologs of FT genes cloned in other species) as the training dataset (Methods, Extended Data Fig. 8 and Supplementary Figs. 13 and 14). The model showed high prediction accuracy and different layers of the interaction map had variable prediction power, with area under the curve (AUC) values ranging from 0.67 to 0.93 (Fig. 5a and Supplementary Fig. 14). We predicted that 2,651 genes are associated with FT (Supplementary Table 12), reflecting the highly complex molecular mechanism underling FT in maize. We assessed the SD between each known validated FT gene and all predicted FT genes. The predicted FT genes had significantly lower SD values (to well-known FT genes) compared to those obtained against background genes (Fig. 5b). Compared to background genes, the predicted FT genes were significantly enriched both in the Arabidopsis homology dataset and in a genome-wide association study of single-nucleotide polymorphisms43 associated with FT(Extended Data Fig. 9a).\n甚至，因為擁有了大量資料，就可以進一步進行預測了。這邊以開花基因作為預測模型。當然，因為開花基因擁有相對較多的true positive，甚至可以參考arabidopsis 資訊 此外，預測完之後，現在也都需要實際knock-out證實預測模型是否正確了 Take home message 對於高度複雜的網路結構問題，作者提出一系列可用的指標，用來拆解網路裡duplicated gene與其功能的關係。也從網路結構裡建立預測模型，用來預測並驗證出一些新穎開花基因\nKeywords network\n","wordCount":"1323","inLanguage":"en","image":"https://xavierzheng.github.io/%3Cimage%20path/url%3E","datePublished":"2022-12-31T08:00:32+08:00","dateModified":"2022-12-31T08:00:32+08:00","author":{"@type":"Person","name":"Me"},"mainEntityOfPage":{"@type":"WebPage","@id":"https://xavierzheng.github.io/posts/maize-network/"},"publisher":{"@type":"Organization","name":"Biology and computer science","logo":{"@type":"ImageObject","url":"https://xavierzheng.github.io/%3Clink%20/%20abs%20url%3E"}}}</script></head><body id=top><script>localStorage.getItem("pref-theme")==="dark"?document.body.classList.add("dark"):localStorage.getItem("pref-theme")==="light"?document.body.classList.remove("dark"):window.matchMedia("(prefers-color-scheme: dark)").matches&&document.body.classList.add("dark")</script><header class=header><nav class=nav><div class=logo><a href=https://xavierzheng.github.io accesskey=h title="Biology and computer science (Alt + H)"><img src=https://xavierzheng.github.io/apple-touch-icon.png alt aria-label=logo height=35>Biology and computer science</a><div class=logo-switches><button id=theme-toggle accesskey=t title="(Alt + T)"><svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24" fill="none" stroke="currentcolor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><path d="M21 12.79A9 9 0 1111.21 3 7 7 0 0021 12.79z"/></svg><svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24" fill="none" stroke="currentcolor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><circle cx="12" cy="12" r="5"/><line x1="12" y1="1" x2="12" y2="3"/><line x1="12" y1="21" x2="12" y2="23"/><line x1="4.22" y1="4.22" x2="5.64" y2="5.64"/><line x1="18.36" y1="18.36" x2="19.78" y2="19.78"/><line x1="1" y1="12" x2="3" y2="12"/><line x1="21" y1="12" x2="23" y2="12"/><line x1="4.22" y1="19.78" x2="5.64" y2="18.36"/><line x1="18.36" y1="5.64" x2="19.78" y2="4.22"/></svg></button></div></div><ul id=menu><li><a href=https://xavierzheng.github.io/search title="search (Alt + /)" accesskey=/><span>search</span></a></li><li><a href=https://xavierzheng.github.io/tags/ title=tags><span>tags</span></a></li><li><a href=https://xavierzheng.github.io/archives/ title=archives><span>archives</span></a></li></ul></nav></header><main class=main><article class=post-single><header class=post-header><div class=breadcrumbs><a href=https://xavierzheng.github.io>Home</a>&nbsp;»&nbsp;<a href=https://xavierzheng.github.io/posts/>Posts</a></div><h1 class=post-title>A multi-omics integrative network map of maize</h1><div class=post-description>如何拆解multi-omics network的好概念，與文章經典格式</div><div class=post-meta><span title='2022-12-31 08:00:32 +0800 CST'>December 31, 2022</span>&nbsp;·&nbsp;Me</div></header><div class=toc><details open><summary accesskey=c title="(Alt + C)"><span class=details>Table of Contents</span></summary><div class=inner><nav id=TableOfContents><ul><li><a href=#ref-a-multi-omics-integrative-network-map-of-maize--nature-geneticshttpswwwnaturecomarticless41588-022-01262-1>ref: <a href=https://www.nature.com/articles/s41588-022-01262-1>A multi-omics integrative network map of maize | Nature Genetics</a></a></li><li><a href=#摘要>摘要</a></li><li><a href=#take-home-message>Take home message</a></li><li><a href=#keywords>Keywords</a></li></ul></nav></div></details></div><div class=post-content><h1 id=念書筆記>念書筆記<a hidden class=anchor aria-hidden=true href=#念書筆記>#</a></h1><h2 id=ref-a-multi-omics-integrative-network-map-of-maize--nature-geneticshttpswwwnaturecomarticless41588-022-01262-1>ref: <a href=https://www.nature.com/articles/s41588-022-01262-1>A multi-omics integrative network map of maize | Nature Genetics</a><a hidden class=anchor aria-hidden=true href=#ref-a-multi-omics-integrative-network-map-of-maize--nature-geneticshttpswwwnaturecomarticless41588-022-01262-1>#</a></h2><p><a href=https://doi.org/10.1038/s41588-022-01262-1>https://doi.org/10.1038/s41588-022-01262-1</a></p><h2 id=摘要>摘要<a hidden class=anchor aria-hidden=true href=#摘要>#</a></h2><blockquote><p><em>Evolution of gene function is often preceded by whole-genome duplication (WGD), tandem duplication (tandem), proximal duplication (proximal) or duplications mediated by transpositions (transposed) and duplications of dispersed origin (dispersed) (Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig2><em>2a</em></a><em>)</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR20><em>20</em></a><em>,</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR21><em>21</em></a><em>. Network modules group genes that potentially participate in the same process. The shortest distance (SD;</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Sec9><em>Methods</em></a><em>) is a metric that represents the shortest path between a pair of nodes within a network. When applied to duplicated genes, a large SD suggests that the two genes are functionally different. All duplicated genes had a significantly higher probability of belonging to the same module (Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig2><em>2b</em></a><em>) and had a lower SD (Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig2><em>2c</em></a><em>) than non-paralogous genes. Tandem and proximal genes were most likely to be enriched in the same network modules, whereas transposed and dispersed genes were least likely. Gene pairs resulting from WGD, tandem and proximal duplications had the lowest SD values, indicating that they are the least functionally divergent, whereas those derived from transposed and dispersed duplications had the highest SD values, suggesting that they are the most functionally divergent (Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig2><em>2b,c</em></a><em>).</em></p></blockquote><ul><li><p>可以把基因功能的演化分成</p><ul><li><p>Whole genome duplication</p></li><li><p>Tandem duplication</p></li><li><p>Proximal duplication</p></li><li><p>Duplication mediated by transposition</p></li><li><p>Dispersed</p></li></ul></li><li><p>在網路結構裡，可以計算兩點間最短距離（short distance)來呈現基因之間的關係。如果最短距離較大，代表兩個基因功能比較不像了。</p></li><li><p>本研究顯示多數duplicate gene還是在同module裏面，而且最短距離也比non paralougs還要短</p></li><li><p>Tandem與proximal通常都在同一個module裡，而transpose比較不會在同一個module</p></li><li><p>以最短距離來比較，whole genome duplication、tandem、proximal通常有較短距離，但transpose、 disperse比較長，代表他們基因功能開始不像了</p></li></ul><blockquote><p><em>To further explore the network divergence of duplicated gene pairs from an evolutionary perspective, we introduced two indices, the Simpson index (SAB) and the divergence index (DAB). The</em> <em>SAB</em> <em>evaluates the similarity between two nodes, while the</em> <em>DAB</em> <em>measures the network divergence of two genes, and these indices are collectively a good indicator of functional divergence (Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig2><em>2d</em></a><em>). Based on</em> <em>SAB</em> <em>and</em> <em>DAB</em> <em>values, we classified all duplicated genes into five divergence patterns (</em><a href=https://www.nature.com/articles/s41588-022-01262-1#Sec9><em>Methods</em></a><em>), with a clear gradual decrease seen in the extent of network similarity between the two genes as differentiation increased from type I to type V (Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig2><em>2d</em></a><em>). These five patterns potentially represent various types of functional evolution: conservation (type I), subfunctionalization (type II), neofunctionalization (types III and IV) and nonfunctionalization/specialization (type V)</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR22><em>22</em></a><em>.</em></p></blockquote><ul><li><p>利用Simpson index (SAB)與divergence index (DAB)可以把所有的duplicate gene拆分成五種。當然這是有reference 參考的分法</p><ul><li><p>conservation (type I),</p></li><li><p>subfunctionalization (type II),</p></li><li><p>neofunctionalization (types III and IV)</p></li><li><p>nonfunctionalization/specialization (type V)</p></li></ul></li></ul><blockquote><p><em>Notably, overall the enrichment ratios gradually decreased from type I to type V compared to background, indicating that duplicated genes share a stronger network similarity than would be expected by chance (Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig2><em>2e</em></a><em>). Different types of duplications, such as WGD and tandem, showed the same trend (Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig2><em>2f</em></a> <em>and Extended Data Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig10><em>5</em></a><em>). Moreover, the proportions of different types of network divergence varied across genetic layers (Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig2><em>2g</em></a> <em>and Extended Data Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig10><em>5</em></a><em>). Interestingly, we observed a dynamic rewiring of networks after gene duplication events from the coexpression to the cotranslation to the interactome level. Our results indicate that approximately 41% of the duplicates were consistent across different omics layer, while approximately 35% were associated with network divergence and approximately 24% were associated with network convergence (Supplementary Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#MOESM1><em>9</em></a><em>). For example, the duplicated genes</em> <em>Zm00001d049815</em> <em>and</em> <em>Zm00001d031611</em> <em>displayed a progressive network rewiring from the coexpression to the cotranslation to the interactome level (Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig2><em>2h</em></a><em>). Importantly, the proportions of duplicated genes originating at different times in the past varied dramatically from type I to V, suggesting that nonfunctionalization, neofunctionalization and specialization are more likely to be associated with older duplications, whereas functional conservation is more likely for the most recent duplications (Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig2><em>2i</em></a><em>). This relationship of network divergence to duplication age became clearer from the coexpression to the cotranslation to the interactome. Together, these results indicate that our integrative network map encompasses potentially evolutionary information relevant to the dissection of gene function in an evolutionary perspective.</em></p></blockquote><ul><li>這種奇異的分類方法之後要接一段敘述，分析利用這種分類方法的優勢及可以看出什麼特徵，再舉個實例證實。當然這篇的優勢在於multiple omits，所有需要將這優勢資料貫穿在分析之中。</li></ul><blockquote><p><em>Variable network divergence of maize subgenomes</em>
<em>Maize is an allotetraploid species with distinct ancient subgenomes, designated here as maize1 and maize2, which exhibit asymmetric divergence in both gene content and expression levels</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR23><em>23</em></a><em>. Although maize1, the dominant subgenome, contains more orthologs and highly expressed genes than maize2, past studies did not find regulating bias, probably because they only assessed this question at a single level</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR23><em>23</em></a><em>,</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR24><em>24</em></a><em>. Therefore, we dissected the regulating bias of the two subgenomes in more detail by classifying all subgenome genes into two groups: maize1 genes with retained maize2 orthologs and maize2 genes with retained maize1 orthologs. We then compared the degree (a network attribute; the number of edges of a node) of genes from each group at different omics levels. To this end, we calculated the variation in the degree of genes from the two different groups. We detected no significant differences between the subgenomes at the coexpression level. The difference increased but was still not significant at the cotranslation level. However, we observed a significant difference at the interactome level between the two subgenomes, as reflected by the degree values (Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig3><em>3a</em></a><em>), which suggests the functional importance of genes and nodes</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR25><em>25</em></a><em>. Additionally, we quantified the overall difference of the degree of genes in a sliding window (100 genes) in the ancient maize subgenomes derived from the</em> <em>Sorghum</em> <em>genome as previously published</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR23><em>23</em></a><em>. We detected more dominant bins for the maize1 subgenome from the coexpression to the cotranslation to the interactome level, suggesting that differences in network connectivity of the subgenomes vary across the three network levels (Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig3><em>3b</em></a> <em>and Supplementary Figs.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#MOESM1><em>10</em></a> <em>and</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#MOESM1><em>11</em></a><em>)</em></p></blockquote><ul><li>植物資料的優勢在於通常有subgenome演化的問題，所有如果能以degree去區分不同subgenome差異，也會是相當有意義。</li></ul><blockquote><p><em>Functional networks of important genes</em>
<em>The integrative network map can reconstruct the functional connections of well-known genes and uncover new regulating genes with similar functions (Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig4><em>4</em></a><em>). We explored a subnetwork involving three key tillering genes,</em> <em>teosinte branched 1</em> <em>(Tb1,</em> <em>Zm00001d033673)</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR26><em>26</em></a><em>,</em> <em>grassy tillers 1</em> <em>(Gt1,</em> <em>Zm00001d028129)</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR27><em>27</em></a> <em>and</em> <em>tassels replace upper ears 1</em> <em>(Tru1,</em> <em>Zm00001d042111)</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR28><em>28</em></a><em>,</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR29><em>29</em></a><em>,</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR30><em>30</em></a><em>,</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR31><em>31</em></a><em>,</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR32><em>32</em></a><em>,</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR33><em>33</em></a><em>,</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR34><em>34</em></a><em>. Loss-of-function mutations in any of these genes lead to the formation of more tillers, a characteristic of the maize ancestor teosinte.</em> <em>Tb1,</em> <em>Gt1</em> <em>and</em> <em>Tru1</em> <em>were all part of a coexpression network with</em> <em>ZmALOG1</em> <em>(Zm00001d003057) and</em> <em>ZmALOG2</em> <em>(Zm00001d032696), two genes of unknown function belonging to the</em> <em>Arabidopsis</em> <em>LSH1 and</em> <em>Oryza</em> <em>G1 (ALOG) transcription factor family (Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig4><em>4a</em></a><em>). Notably, the loss-of-function mutations in</em> <em>ZmALOG1</em> <em>and</em> <em>ZmALOG2</em> <em>obtained by CRISPR- or CRISPR</em>*–*<em>Cas9-mediated gene editing led to higher tillering, similar to the</em> <em>tb1,</em> <em>gt1</em> <em>and</em> <em>tru1</em> <em>mutants (Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig4><em>4b</em></a> <em>and Extended Data Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig11><em>6a</em></a><em>). Moreover, yeast two-hybrid assays indicated that</em> <em>ZmALOG1</em> <em>and</em> <em>ZmALOG2</em> <em>can interact with</em> <em>TB1</em> <em>(Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig4><em>4c</em></a><em>). These results demonstrate that the network edges between known and unknown genes in subnetworks have potential biological meaning. Therefore, our maize network map can be used to generate hypotheses about the functions of genes of interest, as well as their putative network pathways.</em></p></blockquote><ul><li>建立network之後，除了串聯已知基因之外，也可以探討未知基因的功能。所以可以靠突變是否有類似功能之類的資訊，以及yeast two hybrid等方法實際證實未知基因的功能。</li></ul><blockquote><p><em>Predicting novel functional genes for FT</em>
<em>FT is an important agronomic trait involved in the adaptation of maize to a wide range of climates worldwide</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR40><em>40</em></a><em>,</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR41><em>41</em></a><em>,</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR42><em>42</em></a><em>. Although many key FT genes in maize have been cloned (Supplementary Table</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#MOESM3><em>11</em></a><em>), much remains to be learned about the genetic control and molecular mechanisms underlying this trait. To explore the potential regulating networks, we constructed a prediction model based on the integrative map and a machine learning method using known FT genes (functionally cloned FT genes in maize and maize homologs of FT genes cloned in other species) as the training dataset (</em><a href=https://www.nature.com/articles/s41588-022-01262-1#Sec9><em>Methods</em></a><em>, Extended Data Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig13><em>8</em></a> <em>and Supplementary Figs.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#MOESM1><em>13</em></a> <em>and</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#MOESM1><em>14</em></a><em>). The model showed high prediction accuracy and different layers of the interaction map had variable prediction power, with area under the curve (AUC) values ranging from 0.67 to 0.93 (Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig5><em>5a</em></a> <em>and Supplementary Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#MOESM1><em>14</em></a><em>). We predicted that 2,651 genes are associated with FT (Supplementary Table</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#MOESM3><em>12</em></a><em>), reflecting the highly complex molecular mechanism underling FT in maize. We assessed the SD between each known validated FT gene and all predicted FT genes. The predicted FT genes had significantly lower SD values (to well-known FT genes) compared to those obtained against background genes (Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig5><em>5b</em></a><em>). Compared to background genes, the predicted FT genes were significantly enriched both in the</em> <em>Arabidopsis</em> <em>homology dataset and in a genome-wide association study of single-nucleotide polymorphisms</em><a href=https://www.nature.com/articles/s41588-022-01262-1#ref-CR43><em>43</em></a> <em>associated with FT(Extended Data Fig.</em> <a href=https://www.nature.com/articles/s41588-022-01262-1#Fig14><em>9a</em></a><em>).</em></p></blockquote><ul><li>甚至，因為擁有了大量資料，就可以進一步進行預測了。這邊以開花基因作為預測模型。當然，因為開花基因擁有相對較多的true positive，甚至可以參考arabidopsis 資訊</li><li>此外，預測完之後，現在也都需要實際knock-out證實預測模型是否正確了</li></ul><h2 id=take-home-message>Take home message<a hidden class=anchor aria-hidden=true href=#take-home-message>#</a></h2><p>對於高度複雜的網路結構問題，作者提出一系列可用的指標，用來拆解網路裡duplicated gene與其功能的關係。也從網路結構裡建立預測模型，用來預測並驗證出一些新穎開花基因</p><h2 id=keywords>Keywords<a hidden class=anchor aria-hidden=true href=#keywords>#</a></h2><p>network</p></div><footer class=post-footer><ul class=post-tags><li><a href=https://xavierzheng.github.io/tags/network/>network</a></li><li><a href=https://xavierzheng.github.io/tags/paper/>paper</a></li></ul><nav class=paginav><a class=prev href=https://xavierzheng.github.io/posts/bottleneck_in_genomics/><span class=title>« Prev</span><br><span>Navigating bottlenecks and trade-offs in genomic data analysis</span></a>
<a class=next href=https://xavierzheng.github.io/posts/epigenetic-clocks-aging-cancer/><span class=title>Next »</span><br><span>Epigenetic clocks, aging, and cancer</span></a></nav></footer></article></main><footer class=footer><span>&copy; 2022 <a href=https://xavierzheng.github.io>Biology and computer science</a></span>
<span>Powered by
<a href=https://gohugo.io/ rel="noopener noreferrer" target=_blank>Hugo</a> &
        <a href=https://github.com/adityatelange/hugo-PaperMod/ rel=noopener target=_blank>PaperMod</a></span></footer><a href=#top aria-label="go to top" title="Go to Top (Alt + G)" class=top-link id=top-link accesskey=g><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 6" fill="currentcolor"><path d="M12 6H0l6-6z"/></svg></a><script>let menu=document.getElementById("menu");menu&&(menu.scrollLeft=localStorage.getItem("menu-scroll-position"),menu.onscroll=function(){localStorage.setItem("menu-scroll-position",menu.scrollLeft)}),document.querySelectorAll('a[href^="#"]').forEach(e=>{e.addEventListener("click",function(e){e.preventDefault();var t=this.getAttribute("href").substr(1);window.matchMedia("(prefers-reduced-motion: reduce)").matches?document.querySelector(`[id='${decodeURIComponent(t)}']`).scrollIntoView():document.querySelector(`[id='${decodeURIComponent(t)}']`).scrollIntoView({behavior:"smooth"}),t==="top"?history.replaceState(null,null," "):history.pushState(null,null,`#${t}`)})})</script><script>var mybutton=document.getElementById("top-link");window.onscroll=function(){document.body.scrollTop>800||document.documentElement.scrollTop>800?(mybutton.style.visibility="visible",mybutton.style.opacity="1"):(mybutton.style.visibility="hidden",mybutton.style.opacity="0")}</script><script>document.getElementById("theme-toggle").addEventListener("click",()=>{document.body.className.includes("dark")?(document.body.classList.remove("dark"),localStorage.setItem("pref-theme","light")):(document.body.classList.add("dark"),localStorage.setItem("pref-theme","dark"))})</script><script>document.querySelectorAll("pre > code").forEach(e=>{const n=e.parentNode.parentNode,t=document.createElement("button");t.classList.add("copy-code"),t.innerHTML="copy";function s(){t.innerHTML="copied!",setTimeout(()=>{t.innerHTML="copy"},2e3)}t.addEventListener("click",t=>{if("clipboard"in navigator){navigator.clipboard.writeText(e.textContent),s();return}const n=document.createRange();n.selectNodeContents(e);const o=window.getSelection();o.removeAllRanges(),o.addRange(n);try{document.execCommand("copy"),s()}catch{}o.removeRange(n)}),n.classList.contains("highlight")?n.appendChild(t):n.parentNode.firstChild==n||(e.parentNode.parentNode.parentNode.parentNode.parentNode.nodeName=="TABLE"?e.parentNode.parentNode.parentNode.parentNode.parentNode.appendChild(t):e.parentNode.appendChild(t))})</script></body><div id=fastSearch><input id=searchInput tabindex=0><ul id=searchResults></ul></div><script src=/js/fuse.basic.min.js></script>
<script src=/js/fastsearch.js></script></html>