<!doctype html><html lang=en dir=auto><head><meta charset=utf-8><meta http-equiv=x-ua-compatible content="IE=edge"><meta name=viewport content="width=device-width,initial-scale=1,shrink-to-fit=no"><meta name=robots content="index, follow"><title>Tips GO | Biology and computer science</title><meta name=keywords content="GO"><meta name=description content="GO注意事項 Ref: Nine quick tips for pathway enrichment analysis, 2022 PLOS computation biology, https://doi.org/10.1371/journal.pcbi.1010348
Tip 1: Before starting, clarify which analysis you would like to perform What analysis type Pathway enrichment analysis (PEA) does not give clues about the active or inhibited status of the pathways. More appropriately, PEA provides information about how genes help carry out pathways.
overrepresentation analysis (ORA) focus of ORA methods is the gene set
ORA outputs all pathways enriched in the query gene list as a whole, and mainly uses a nonranked list"><meta name=author content="Me"><link rel=canonical href=https://canonical.url/to/page><link crossorigin=anonymous href=/assets/css/stylesheet.bccfefac377bc340f06c260aed1bddf49a4354816d7c570d6aac75a997986c95.css integrity="sha256-vM/vrDd7w0DwbCYK7Rvd9JpDVIFtfFcNaqx1qZeYbJU=" rel="preload stylesheet" as=style><script defer crossorigin=anonymous src=/assets/js/highlight.f413e19d0714851f6474e7ee9632408e58ac146fbdbe62747134bea2fa3415e0.js integrity="sha256-9BPhnQcUhR9kdOfuljJAjlisFG+9vmJ0cTS+ovo0FeA=" onload=hljs.initHighlightingOnLoad()></script>
<link rel=icon href=https://xavierzheng.github.io/%3Clink%20/%20abs%20url%3E><link rel=icon type=image/png sizes=16x16 href=https://xavierzheng.github.io/%3Clink%20/%20abs%20url%3E><link rel=icon type=image/png sizes=32x32 href=https://xavierzheng.github.io/%3Clink%20/%20abs%20url%3E><link rel=apple-touch-icon href=https://xavierzheng.github.io/%3Clink%20/%20abs%20url%3E><link rel=mask-icon href=https://xavierzheng.github.io/%3Clink%20/%20abs%20url%3E><meta name=theme-color content="#2e2e33"><meta name=msapplication-TileColor content="#2e2e33"><noscript><style>#theme-toggle,.top-link{display:none}</style><style>@media(prefers-color-scheme:dark){:root{--theme:rgb(29, 30, 32);--entry:rgb(46, 46, 51);--primary:rgb(218, 218, 219);--secondary:rgb(155, 156, 157);--tertiary:rgb(65, 66, 68);--content:rgb(196, 196, 197);--hljs-bg:rgb(46, 46, 51);--code-bg:rgb(55, 56, 62);--border:rgb(51, 51, 51)}.list{background:var(--theme)}.list:not(.dark)::-webkit-scrollbar-track{background:0 0}.list:not(.dark)::-webkit-scrollbar-thumb{border-color:var(--theme)}}</style></noscript><meta property="og:title" content="Tips GO"><meta property="og:description" content="GO注意事項 Ref: Nine quick tips for pathway enrichment analysis, 2022 PLOS computation biology, https://doi.org/10.1371/journal.pcbi.1010348
Tip 1: Before starting, clarify which analysis you would like to perform What analysis type Pathway enrichment analysis (PEA) does not give clues about the active or inhibited status of the pathways. More appropriately, PEA provides information about how genes help carry out pathways.
overrepresentation analysis (ORA) focus of ORA methods is the gene set
ORA outputs all pathways enriched in the query gene list as a whole, and mainly uses a nonranked list"><meta property="og:type" content="article"><meta property="og:url" content="https://xavierzheng.github.io/posts/tips-go/"><meta property="og:image" content="https://xavierzheng.github.io/%3Cimage%20path/url%3E"><meta property="article:section" content="posts"><meta property="article:published_time" content="2022-12-28T11:51:48+08:00"><meta property="article:modified_time" content="2022-12-28T11:51:48+08:00"><meta name=twitter:card content="summary_large_image"><meta name=twitter:image content="https://xavierzheng.github.io/%3Cimage%20path/url%3E"><meta name=twitter:title content="Tips GO"><meta name=twitter:description content="GO注意事項 Ref: Nine quick tips for pathway enrichment analysis, 2022 PLOS computation biology, https://doi.org/10.1371/journal.pcbi.1010348
Tip 1: Before starting, clarify which analysis you would like to perform What analysis type Pathway enrichment analysis (PEA) does not give clues about the active or inhibited status of the pathways. More appropriately, PEA provides information about how genes help carry out pathways.
overrepresentation analysis (ORA) focus of ORA methods is the gene set
ORA outputs all pathways enriched in the query gene list as a whole, and mainly uses a nonranked list"><script type=application/ld+json>{"@context":"https://schema.org","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","position":2,"name":"Posts","item":"https://xavierzheng.github.io/posts/"},{"@type":"ListItem","position":3,"name":"Tips GO","item":"https://xavierzheng.github.io/posts/tips-go/"}]}</script><script type=application/ld+json>{"@context":"https://schema.org","@type":"BlogPosting","headline":"Tips GO","name":"Tips GO","description":"GO注意事項 Ref: Nine quick tips for pathway enrichment analysis, 2022 PLOS computation biology, https://doi.org/10.1371/journal.pcbi.1010348\nTip 1: Before starting, clarify which analysis you would like to perform What analysis type Pathway enrichment analysis (PEA) does not give clues about the active or inhibited status of the pathways. More appropriately, PEA provides information about how genes help carry out pathways.\noverrepresentation analysis (ORA) focus of ORA methods is the gene set\nORA outputs all pathways enriched in the query gene list as a whole, and mainly uses a nonranked list","keywords":["GO"],"articleBody":"GO注意事項 Ref: Nine quick tips for pathway enrichment analysis, 2022 PLOS computation biology, https://doi.org/10.1371/journal.pcbi.1010348\nTip 1: Before starting, clarify which analysis you would like to perform What analysis type Pathway enrichment analysis (PEA) does not give clues about the active or inhibited status of the pathways. More appropriately, PEA provides information about how genes help carry out pathways.\noverrepresentation analysis (ORA) focus of ORA methods is the gene set\nORA outputs all pathways enriched in the query gene list as a whole, and mainly uses a nonranked list\n(except one option in g:Profiler g:GOst using a minimum hypergeometric value-based method).\ngene set enrichment analysis (GSEA) approaches 代表某一個pathway在transcriptome資料裡面有enrich focus of GSEA techniques is the ranked pathways list it is advisable to choose GSEA methods when there is uncertainty about the cutoff value. GSEA indicates the pathways that are enriched in genes located at both extreme ends of a ranked gene list, and a higher ranked pathway indicates that more genes are located at the very top or at the very end of this list competitive methods: BioPAX-Parser (BiP) [35] pathDIP [36,37] SPIA [38] CePaORA [39] PathNet [40] self-contained methods: CePa [39] GSEA [25–27] Which data type For unordered lists of genes, researchers can use g:Profiler g:GOSt [8–10], Enrichr [28,29], and BioPAX-Parser [35,47]. If the genes are ranked, g:Profiler g:GOSt can treasure this information and generate rank-based functional enrichment results. **If the input data are gene expression levels, they can be analyzed through GSEA [27]. ** 如果是expression level, 可以直接用GSEA pathDIP [37], instead, can assist with curated analyses based on scientific literature. Tip 2: Ensure the quality of your input genes or genomic regions 請確定放進去的是對的ID，有意義的list，有意義的expression\nTip 3: Use multiple PEA tools, not only one we suggest all the practitioners performing a functional enrichment analysis phase to employ at least two different PEA tools\nif the user had an unranked list of gene symbols, we would suggest to apply g:Profiler g:GOSt [10], Enrichr [28], and GeneTrail [52] to it, and then compare their results. To perform pathway enrichment by employing more than a single database, users can employ cPEA [48], a software tool able to deal with several pathway databases using the BioPAX language [61] to store and represent pathways. Or they can use BiP [47] by selecting the “Whole PathwayCommon Data” option that will perform cross enrichment using the whole collection of automatically downloaded locally Pathway Common databases [62] coded in BioPAX. Tip 4: Document all your PEA tests and their details 記得要寫好筆記，所有使用的參數以及資料庫的版本都要清楚寫下來，這是dry lab的實驗紀錄\nBox 1: Example of PEA test details My test ID: 2022-02-04, h10:02 EST. My input genes: AK4, ALDOC, EGLN1, FAM162A, MTFP1, PDK1, PGK1. My input genes’ type: gene symbols. Source: D. Cangelosi and colleagues [69]. Disease: neuroblastoma. Tool: g:Profiler g:GOSt. Access: online via Google Chrome browser. Version: e104_eg51_p15_3922dba. URL: https://biit.cs.ut.ee/gprofiler/gost Organism: Homo sapiens. Query: unordered genes. Statistical domain scope: only annotated genes. Significance threshold: g:SCS threshold. User threshold: 0.005. Data sources: default. All the other parameters: default. My output file(s) name(s): gprofiler_gost_NB_2022-02-04_h1002_output.csv My output file(s) folder: /home/davide/PEA_analyses/neuroblastoma/ My output file(s) location: bioinformatics-laptop-2021 (Dell Latitude E5420). Tip 5: Always use the corrected p-value, and not the nominal one terms adjusted p-values, corrected p-values, and false discovery rate (FDR) values are often used as synonyms in the scientific literature we suggest using the adjusted p-value threshold at 0.005 (that corresponds to 5 × 10−3), as recommended by Benjamin and colleagues [75]. We therefore suggest using the p.adj \u003c 0.005 threshold for a first strict analysis of the results, and then repeating the test by using a more permissive threshold such as p.adj \u003c 0.01, and then again with an even higher threshold, such as the traditional p.adj \u003c 0.05. Based on the characteristics of the experiment, results found by one particular threshold might be more suitable than results found with other thresholds. 記得從嚴格到寬鬆的原則 Tip 6: Keep in mind that your PEA results can be strongly affected by the statistical tests and the visualization techniques you use Statistical tests. 統計方法不一樣，會得到不一樣的結果 沒有最適合的方法，只有適合你實驗設計的方法 Visualization Enrichment Maps [80] and enrichplot [81] for biological pathways, AutoAnnotate [82] for networks, REVIGO [83] and CirGO [84] for GO annotations are few examples of different visualization techniques and contents. Network visualization techniques can also be used to detect a lower adjusted p-value threshold (Tip 5). Tip 7: Consider using subgroups of correlated genes instead of all your input genes Instead of using all genes as input for the PEA, we therefore suggest bioinformatics practitioners to detect subgroups of correlated genes and perform the PEA on each of these subgroups alone.\n不要一口氣全部做，可以考慮subgroup\nSubgroups of correlated genes can be found, for example, through protein–protein interaction networks’ tools such as IID [85–88], STRING [89–92], GeneMANIA [93–97], or Reactome Functional Interaction Network (Reactome FI) [98,99]. some R packages have been recently released: pathfindR [102] and netGO [103], which exploit the protein–protein interaction networks to produce more accurate PEA results. Tip 8: Use the (recent) scientific literature to review your PEA results we therefore suggest any practitioner to manually perform a literature search and look for scientific studies published about the significant genes–pathways associations found by the functional enrichment analysis and about the role of the genes inside the enriched pathways found.\n請找paper確定文獻是否更新\nTip 9: Ask a wet lab biologist or a clinician to review your PEA results a wet lab biologist or a clinician should review these results and clearly say if they make sense or if they contain mistakes or inappropriate information.\n請找專家用眼睛看\n","wordCount":"901","inLanguage":"en","image":"https://xavierzheng.github.io/%3Cimage%20path/url%3E","datePublished":"2022-12-28T11:51:48+08:00","dateModified":"2022-12-28T11:51:48+08:00","author":{"@type":"Person","name":"Me"},"mainEntityOfPage":{"@type":"WebPage","@id":"https://xavierzheng.github.io/posts/tips-go/"},"publisher":{"@type":"Organization","name":"Biology and computer science","logo":{"@type":"ImageObject","url":"https://xavierzheng.github.io/%3Clink%20/%20abs%20url%3E"}}}</script></head><body id=top><script>localStorage.getItem("pref-theme")==="dark"?document.body.classList.add("dark"):localStorage.getItem("pref-theme")==="light"?document.body.classList.remove("dark"):window.matchMedia("(prefers-color-scheme: dark)").matches&&document.body.classList.add("dark")</script><header class=header><nav class=nav><div class=logo><a href=https://xavierzheng.github.io accesskey=h title="Biology and computer science (Alt + H)"><img src=https://xavierzheng.github.io/apple-touch-icon.png alt aria-label=logo height=35>Biology and computer science</a><div class=logo-switches><button id=theme-toggle accesskey=t title="(Alt + T)"><svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24" fill="none" stroke="currentcolor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><path d="M21 12.79A9 9 0 1111.21 3 7 7 0 0021 12.79z"/></svg><svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24" fill="none" stroke="currentcolor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><circle cx="12" cy="12" r="5"/><line x1="12" y1="1" x2="12" y2="3"/><line x1="12" y1="21" x2="12" y2="23"/><line x1="4.22" y1="4.22" x2="5.64" y2="5.64"/><line x1="18.36" y1="18.36" x2="19.78" y2="19.78"/><line x1="1" y1="12" x2="3" y2="12"/><line x1="21" y1="12" x2="23" y2="12"/><line x1="4.22" y1="19.78" x2="5.64" y2="18.36"/><line x1="18.36" y1="5.64" x2="19.78" y2="4.22"/></svg></button></div></div><ul id=menu><li><a href=https://xavierzheng.github.io/search/ title="search (Alt + /)" accesskey=/><span>search</span></a></li><li><a href=https://xavierzheng.github.io/tags/ title=tags><span>tags</span></a></li><li><a href=https://xavierzheng.github.io/archives/ title=archives><span>archives</span></a></li></ul></nav></header><main class=main><article class=post-single><header class=post-header><div class=breadcrumbs><a href=https://xavierzheng.github.io>Home</a>&nbsp;»&nbsp;<a href=https://xavierzheng.github.io/posts/>Posts</a></div><h1 class=post-title>Tips GO</h1><div class=post-meta><span title='2022-12-28 11:51:48 +0800 CST'>December 28, 2022</span>&nbsp;·&nbsp;Me</div></header><div class=toc><details open><summary accesskey=c title="(Alt + C)"><span class=details>Table of Contents</span></summary><div class=inner><nav id=TableOfContents><ul><li><a href=#tip-1-before-starting-clarify-which-analysis-you-would-like-to-perform>Tip 1: Before starting, clarify which analysis you would like to perform</a><ul><li><a href=#what-analysis-type>What analysis type</a></li><li><a href=#which-data-type>Which data type</a></li></ul></li><li><a href=#tip-2-ensure-the-quality-of-your-input-genes-or-genomic-regions>Tip 2: Ensure the quality of your input genes or genomic regions</a></li><li><a href=#tip-3-use-multiple-pea-tools-not-only-one>Tip 3: Use multiple PEA tools, not only one</a></li><li><a href=#tip-4-document-all-your-pea-tests-and-their-details>Tip 4: Document all your PEA tests and their details</a></li><li><a href=#tip-5-always-use-the-corrected-p-value-and-not-the-nominal-one>Tip 5: Always use the corrected <em>p</em>-value, and not the nominal one</a></li><li><a href=#tip-6-keep-in-mind-that-your-pea-results-can-be-strongly-affected--by-the-statistical-tests-and-the-visualization-techniques-you-use>Tip 6: Keep in mind that your PEA results can be strongly affected by the statistical tests and the visualization techniques you use</a><ul><li><a href=#statistical-tests>Statistical tests.</a></li><li><a href=#visualization>Visualization</a></li></ul></li><li><a href=#tip-7-consider-using-subgroups-of-correlated-genes-instead-of-all-your-input-genes>Tip 7: Consider using subgroups of correlated genes instead of all your input genes</a></li><li><a href=#tip-8-use-the-recent-scientific-literature-to-review-your-pea-results>Tip 8: Use the (recent) scientific literature to review your PEA results</a></li><li><a href=#tip-9-ask-a-wet-lab-biologist-or-a-clinician-to-review-your-pea-results>Tip 9: Ask a wet lab biologist or a clinician to review your PEA results</a></li></ul></nav></div></details></div><div class=post-content><h1 id=go注意事項>GO注意事項<a hidden class=anchor aria-hidden=true href=#go注意事項>#</a></h1><p><strong>Ref</strong>: Nine quick tips for pathway enrichment analysis, 2022 PLOS computation biology, <a href=https://doi.org/10.1371/journal.pcbi.1010348>https://doi.org/10.1371/journal.pcbi.1010348</a></p><h2 id=tip-1-before-starting-clarify-which-analysis-you-would-like-to-perform>Tip 1: Before starting, clarify which analysis you would like to perform<a hidden class=anchor aria-hidden=true href=#tip-1-before-starting-clarify-which-analysis-you-would-like-to-perform>#</a></h2><h3 id=what-analysis-type>What analysis type<a hidden class=anchor aria-hidden=true href=#what-analysis-type>#</a></h3><p>Pathway enrichment analysis (PEA) does not give clues about the active or inhibited status of the pathways. More appropriately, PEA provides information about how genes help carry out pathways.</p><h4 id=overrepresentation-analysis-ora>overrepresentation analysis (ORA)<a hidden class=anchor aria-hidden=true href=#overrepresentation-analysis-ora>#</a></h4><ul><li><p><strong>focus of ORA methods is the gene set</strong></p></li><li><p>ORA outputs all pathways enriched in the query gene list as a whole, and mainly uses a nonranked list</p></li><li><p>(except one option in g:Profiler g:GOst using a minimum hypergeometric value-based method).</p></li></ul><h4 id=gene-set-enrichment-analysis-gsea-approaches>gene set enrichment analysis (GSEA) approaches<a hidden class=anchor aria-hidden=true href=#gene-set-enrichment-analysis-gsea-approaches>#</a></h4><ul><li>代表某一個pathway在transcriptome資料裡面有enrich</li><li><strong>focus of GSEA techniques is the ranked pathways list</strong></li><li>it is advisable to choose GSEA methods when there is uncertainty about the cutoff value.</li><li>GSEA indicates the pathways that are enriched in genes located at both extreme ends of a ranked gene list, and a higher ranked pathway indicates that more genes are located at the very top or at the very end of this list<ul><li>competitive methods:</li><li>BioPAX-Parser (BiP) [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref035">35</a>]</li><li>pathDIP [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref036">36</a>,<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref037">37</a>]</li><li>SPIA [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref038">38</a>]</li><li>CePaORA [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref039">39</a>]</li><li>PathNet [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref040">40</a>]</li><li>self-contained methods:</li><li>CePa [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref039">39</a>]</li><li>GSEA [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref025">25</a>–<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref027">27</a>]</li></ul></li></ul><h3 id=which-data-type>Which data type<a hidden class=anchor aria-hidden=true href=#which-data-type>#</a></h3><ul><li>For unordered lists of genes, researchers can use g:Profiler g:GOSt [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref008">8</a>–<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref010">10</a>], Enrichr [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref028">28</a>,<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref029">29</a>], and BioPAX-Parser [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref035">35</a>,<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref047">47</a>].</li><li>If the genes are ranked, g:Profiler g:GOSt can treasure this information and generate rank-based functional enrichment results.</li><li>**If the input data are gene expression levels, they can be analyzed through GSEA [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref027">27</a>]. **</li><li><strong>如果是expression level, 可以直接用GSEA</strong></li><li>pathDIP [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref037">37</a>], instead, can assist with curated analyses based on scientific literature.</li></ul><h2 id=tip-2-ensure-the-quality-of-your-input-genes-or-genomic-regions>Tip 2: Ensure the quality of your input genes or genomic regions<a hidden class=anchor aria-hidden=true href=#tip-2-ensure-the-quality-of-your-input-genes-or-genomic-regions>#</a></h2><p>請確定放進去的是對的ID，有意義的list，有意義的expression</p><h2 id=tip-3-use-multiple-pea-tools-not-only-one>Tip 3: Use multiple PEA tools, not only one<a hidden class=anchor aria-hidden=true href=#tip-3-use-multiple-pea-tools-not-only-one>#</a></h2><p>we suggest all the practitioners performing a functional enrichment analysis phase to employ at least two different PEA tools</p><ul><li>if the user had an unranked list of gene symbols, we would suggest to apply g:Profiler g:GOSt [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref010">10</a>], Enrichr [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref028">28</a>], and GeneTrail [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref052">52</a>] to it, and then compare their results.</li><li>To perform pathway enrichment by employing more than a single database, users can employ cPEA [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref048">48</a>], a software tool able to deal with several pathway databases using the BioPAX language [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref061">61</a>] to store and represent pathways.</li><li>Or they can use BiP [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref047">47</a>] by selecting the “Whole PathwayCommon Data” option that will perform cross enrichment using the whole collection of automatically downloaded locally Pathway Common databases [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref062">62</a>] coded in BioPAX.</li></ul><h2 id=tip-4-document-all-your-pea-tests-and-their-details>Tip 4: Document all your PEA tests and their details<a hidden class=anchor aria-hidden=true href=#tip-4-document-all-your-pea-tests-and-their-details>#</a></h2><p>記得要寫好筆記，所有使用的參數以及資料庫的版本都要清楚寫下來，這是dry lab的實驗紀錄</p><pre tabindex=0><code>Box 1: Example of PEA test details

My test ID: 2022-02-04, h10:02 EST.

My input genes: AK4, ALDOC, EGLN1, FAM162A, MTFP1, PDK1, PGK1.

My input genes’ type: gene symbols.

Source: D. Cangelosi and colleagues [69].

Disease: neuroblastoma.

Tool: g:Profiler g:GOSt.

Access: online via Google Chrome browser.

Version: e104_eg51_p15_3922dba.

URL: https://biit.cs.ut.ee/gprofiler/gost

Organism: Homo sapiens.

Query: unordered genes.

Statistical domain scope: only annotated genes.

Significance threshold: g:SCS threshold.

User threshold: 0.005.

Data sources: default.

All the other parameters: default.

My output file(s) name(s): gprofiler_gost_NB_2022-02-04_h1002_output.csv

My output file(s) folder: /home/davide/PEA_analyses/neuroblastoma/

My output file(s) location: bioinformatics-laptop-2021 (Dell Latitude E5420).
</code></pre><h2 id=tip-5-always-use-the-corrected-p-value-and-not-the-nominal-one>Tip 5: Always use the corrected <em>p</em>-value, and not the nominal one<a hidden class=anchor aria-hidden=true href=#tip-5-always-use-the-corrected-p-value-and-not-the-nominal-one>#</a></h2><ul><li>terms adjusted p-values, corrected p-values, and false discovery rate (FDR) values are often used as synonyms in the scientific literature</li><li>we suggest using the adjusted <em>p</em>-value threshold at 0.005 (that corresponds to 5 × 10−3), as recommended by Benjamin and colleagues [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref075">75</a>].</li><li>We therefore suggest using the <em>p</em>.<em>adj</em> &lt; 0.005 threshold for a first strict analysis of the results, and then repeating the test by using a more permissive threshold such as <em>p</em>.<em>adj</em> &lt; 0.01, and then again with an even higher threshold, such as the traditional <em>p</em>.<em>adj</em> &lt; 0.05. Based on the characteristics of the experiment, results found by one particular threshold might be more suitable than results found with other thresholds.</li><li>記得從嚴格到寬鬆的原則</li></ul><h2 id=tip-6-keep-in-mind-that-your-pea-results-can-be-strongly-affected--by-the-statistical-tests-and-the-visualization-techniques-you-use>Tip 6: Keep in mind that your PEA results can be strongly affected by the statistical tests and the visualization techniques you use<a hidden class=anchor aria-hidden=true href=#tip-6-keep-in-mind-that-your-pea-results-can-be-strongly-affected--by-the-statistical-tests-and-the-visualization-techniques-you-use>#</a></h2><h3 id=statistical-tests>Statistical tests.<a hidden class=anchor aria-hidden=true href=#statistical-tests>#</a></h3><ul><li>統計方法不一樣，會得到不一樣的結果</li><li>沒有最適合的方法，只有適合你實驗設計的方法</li></ul><h3 id=visualization>Visualization<a hidden class=anchor aria-hidden=true href=#visualization>#</a></h3><ul><li>Enrichment Maps [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref080">80</a>] and enrichplot [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref081">81</a>] for biological pathways,</li><li>AutoAnnotate [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref082">82</a>] for networks,</li><li>REVIGO [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref083">83</a>] and CirGO [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref084">84</a>] for GO annotations are few examples of different visualization techniques and contents.</li><li>Network visualization techniques can also be used to detect a lower adjusted <em>p</em>-value threshold (Tip 5).</li></ul><h2 id=tip-7-consider-using-subgroups-of-correlated-genes-instead-of-all-your-input-genes>Tip 7: Consider using subgroups of correlated genes instead of all your input genes<a hidden class=anchor aria-hidden=true href=#tip-7-consider-using-subgroups-of-correlated-genes-instead-of-all-your-input-genes>#</a></h2><p>Instead of using all genes as input for the PEA, we therefore suggest bioinformatics practitioners to detect subgroups of correlated genes and perform the PEA on each of these subgroups alone.</p><p>不要一口氣全部做，可以考慮subgroup</p><ul><li>Subgroups of correlated genes can be found, for example, through protein–protein interaction networks’ tools such as IID [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref085">85</a>–<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref088">88</a>], STRING [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref089">89</a>–<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref092">92</a>], GeneMANIA [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref093">93</a>–<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref097">97</a>], or Reactome Functional Interaction Network (Reactome FI) [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref098">98</a>,<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref099">99</a>].</li><li>some R packages have been recently released: <strong>pathfindR [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref102">102</a>] and netGO [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref103">103</a>], which exploit the protein–protein interaction networks to produce more accurate PEA results.</strong></li></ul><h2 id=tip-8-use-the-recent-scientific-literature-to-review-your-pea-results>Tip 8: Use the (recent) scientific literature to review your PEA results<a hidden class=anchor aria-hidden=true href=#tip-8-use-the-recent-scientific-literature-to-review-your-pea-results>#</a></h2><p>we therefore suggest any practitioner to manually perform a literature search and look for scientific studies published about the significant genes–pathways associations found by the functional enrichment analysis and about the role of the genes inside the enriched pathways found.</p><p>請找paper確定文獻是否更新</p><h2 id=tip-9-ask-a-wet-lab-biologist-or-a-clinician-to-review-your-pea-results>Tip 9: Ask a wet lab biologist or a clinician to review your PEA results<a hidden class=anchor aria-hidden=true href=#tip-9-ask-a-wet-lab-biologist-or-a-clinician-to-review-your-pea-results>#</a></h2><p>a wet lab biologist or a clinician should review these results and clearly say if they make sense or if they contain mistakes or inappropriate information.</p><p>請找專家用眼睛看</p></div><footer class=post-footer><ul class=post-tags><li><a href=https://xavierzheng.github.io/tags/go/>GO</a></li></ul><nav class=paginav><a class=prev href=https://xavierzheng.github.io/posts/%E7%94%A8hugo%E5%92%8Cgithub%E8%A3%BD%E4%BD%9C%E9%9D%9C%E6%85%8B%E7%B6%B2%E7%AB%99/><span class=title>« Prev</span><br><span>用HUGO和github製作靜態網站</span></a></nav><div class=share-buttons><a target=_blank rel="noopener noreferrer" aria-label="share Tips GO on twitter" href="https://twitter.com/intent/tweet/?text=Tips%20GO&url=https%3a%2f%2fxavierzheng.github.io%2fposts%2ftips-go%2f&hashtags=GO"><svg viewBox="0 0 512 512" height="30" width="30" fill="currentcolor"><path d="M449.446.0C483.971.0 512 28.03 512 62.554v386.892C512 483.97 483.97 512 449.446 512H62.554c-34.524.0-62.554-28.03-62.554-62.554V62.554c0-34.524 28.029-62.554 62.554-62.554h386.892zM195.519 424.544c135.939.0 210.268-112.643 210.268-210.268.0-3.218.0-6.437-.153-9.502 14.406-10.421 26.973-23.448 36.935-38.314-13.18 5.824-27.433 9.809-42.452 11.648 15.326-9.196 26.973-23.602 32.49-40.92-14.252 8.429-30.038 14.56-46.896 17.931-13.487-14.406-32.644-23.295-53.946-23.295-40.767.0-73.87 33.104-73.87 73.87.0 5.824.613 11.494 1.992 16.858-61.456-3.065-115.862-32.49-152.337-77.241-6.284 10.881-9.962 23.601-9.962 37.088.0 25.594 13.027 48.276 32.95 61.456-12.107-.307-23.448-3.678-33.41-9.196v.92c0 35.862 25.441 65.594 59.311 72.49-6.13 1.686-12.72 2.606-19.464 2.606-4.751.0-9.348-.46-13.946-1.38 9.349 29.426 36.628 50.728 68.965 51.341-25.287 19.771-57.164 31.571-91.8 31.571-5.977.0-11.801-.306-17.625-1.073 32.337 21.15 71.264 33.41 112.95 33.41z"/></svg></a><a target=_blank rel="noopener noreferrer" aria-label="share Tips GO on linkedin" href="https://www.linkedin.com/shareArticle?mini=true&url=https%3a%2f%2fxavierzheng.github.io%2fposts%2ftips-go%2f&title=Tips%20GO&summary=Tips%20GO&source=https%3a%2f%2fxavierzheng.github.io%2fposts%2ftips-go%2f"><svg viewBox="0 0 512 512" height="30" width="30" fill="currentcolor"><path d="M449.446.0C483.971.0 512 28.03 512 62.554v386.892C512 483.97 483.97 512 449.446 512H62.554c-34.524.0-62.554-28.03-62.554-62.554V62.554c0-34.524 28.029-62.554 62.554-62.554h386.892zM160.461 423.278V197.561h-75.04v225.717h75.04zm270.539.0V293.839c0-69.333-37.018-101.586-86.381-101.586-39.804.0-57.634 21.891-67.617 37.266v-31.958h-75.021c.995 21.181.0 225.717.0 225.717h75.02V297.222c0-6.748.486-13.492 2.474-18.315 5.414-13.475 17.767-27.434 38.494-27.434 27.135.0 38.007 20.707 38.007 51.037v120.768H431zM123.448 88.722C97.774 88.722 81 105.601 81 127.724c0 21.658 16.264 39.002 41.455 39.002h.484c26.165.0 42.452-17.344 42.452-39.002-.485-22.092-16.241-38.954-41.943-39.002z"/></svg></a><a target=_blank rel="noopener noreferrer" aria-label="share Tips GO on reddit" href="https://reddit.com/submit?url=https%3a%2f%2fxavierzheng.github.io%2fposts%2ftips-go%2f&title=Tips%20GO"><svg viewBox="0 0 512 512" height="30" width="30" fill="currentcolor"><path d="M449.446.0C483.971.0 512 28.03 512 62.554v386.892C512 483.97 483.97 512 449.446 512H62.554c-34.524.0-62.554-28.03-62.554-62.554V62.554c0-34.524 28.029-62.554 62.554-62.554h386.892zM446 265.638c0-22.964-18.616-41.58-41.58-41.58-11.211.0-21.361 4.457-28.841 11.666-28.424-20.508-67.586-33.757-111.204-35.278l18.941-89.121 61.884 13.157c.756 15.734 13.642 28.29 29.56 28.29 16.407.0 29.706-13.299 29.706-29.701.0-16.403-13.299-29.702-29.706-29.702-11.666.0-21.657 6.792-26.515 16.578l-69.105-14.69c-1.922-.418-3.939-.042-5.585 1.036-1.658 1.073-2.811 2.761-3.224 4.686l-21.152 99.438c-44.258 1.228-84.046 14.494-112.837 35.232-7.468-7.164-17.589-11.591-28.757-11.591-22.965.0-41.585 18.616-41.585 41.58.0 16.896 10.095 31.41 24.568 37.918-.639 4.135-.99 8.328-.99 12.576.0 63.977 74.469 115.836 166.33 115.836s166.334-51.859 166.334-115.836c0-4.218-.347-8.387-.977-12.493 14.564-6.47 24.735-21.034 24.735-38.001zM326.526 373.831c-20.27 20.241-59.115 21.816-70.534 21.816-11.428.0-50.277-1.575-70.522-21.82-3.007-3.008-3.007-7.882.0-10.889 3.003-2.999 7.882-3.003 10.885.0 12.777 12.781 40.11 17.317 59.637 17.317 19.522.0 46.86-4.536 59.657-17.321 3.016-2.999 7.886-2.995 10.885.008 3.008 3.011 3.003 7.882-.008 10.889zm-5.23-48.781c-16.373.0-29.701-13.324-29.701-29.698.0-16.381 13.328-29.714 29.701-29.714 16.378.0 29.706 13.333 29.706 29.714.0 16.374-13.328 29.698-29.706 29.698zM160.91 295.348c0-16.381 13.328-29.71 29.714-29.71 16.369.0 29.689 13.329 29.689 29.71.0 16.373-13.32 29.693-29.689 29.693-16.386.0-29.714-13.32-29.714-29.693z"/></svg></a><a target=_blank rel="noopener noreferrer" aria-label="share Tips GO on facebook" href="https://facebook.com/sharer/sharer.php?u=https%3a%2f%2fxavierzheng.github.io%2fposts%2ftips-go%2f"><svg viewBox="0 0 512 512" height="30" width="30" fill="currentcolor"><path d="M449.446.0C483.971.0 512 28.03 512 62.554v386.892C512 483.97 483.97 512 449.446 512H342.978V319.085h66.6l12.672-82.621h-79.272v-53.617c0-22.603 11.073-44.636 46.58-44.636H425.6v-70.34s-32.71-5.582-63.982-5.582c-65.288.0-107.96 39.569-107.96 111.204v62.971h-72.573v82.621h72.573V512h-191.104c-34.524.0-62.554-28.03-62.554-62.554V62.554c0-34.524 28.029-62.554 62.554-62.554h386.892z"/></svg></a><a target=_blank rel="noopener noreferrer" aria-label="share Tips GO on whatsapp" href="https://api.whatsapp.com/send?text=Tips%20GO%20-%20https%3a%2f%2fxavierzheng.github.io%2fposts%2ftips-go%2f"><svg viewBox="0 0 512 512" height="30" width="30" fill="currentcolor"><path d="M449.446.0C483.971.0 512 28.03 512 62.554v386.892C512 483.97 483.97 512 449.446 512H62.554c-34.524.0-62.554-28.03-62.554-62.554V62.554c0-34.524 28.029-62.554 62.554-62.554h386.892zm-58.673 127.703c-33.842-33.881-78.847-52.548-126.798-52.568-98.799.0-179.21 80.405-179.249 179.234-.013 31.593 8.241 62.428 23.927 89.612l-25.429 92.884 95.021-24.925c26.181 14.28 55.659 21.807 85.658 21.816h.074c98.789.0 179.206-80.413 179.247-179.243.018-47.895-18.61-92.93-52.451-126.81zM263.976 403.485h-.06c-26.734-.01-52.954-7.193-75.828-20.767l-5.441-3.229-56.386 14.792 15.05-54.977-3.542-5.637c-14.913-23.72-22.791-51.136-22.779-79.287.033-82.142 66.867-148.971 149.046-148.971 39.793.014 77.199 15.531 105.329 43.692 28.128 28.16 43.609 65.592 43.594 105.4-.034 82.149-66.866 148.983-148.983 148.984zm81.721-111.581c-4.479-2.242-26.499-13.075-30.604-14.571-4.105-1.495-7.091-2.241-10.077 2.241-2.986 4.483-11.569 14.572-14.182 17.562-2.612 2.988-5.225 3.364-9.703 1.12-4.479-2.241-18.91-6.97-36.017-22.23C231.8 264.15 222.81 249.484 220.198 245s-.279-6.908 1.963-9.14c2.016-2.007 4.48-5.232 6.719-7.847 2.24-2.615 2.986-4.484 4.479-7.472 1.493-2.99.747-5.604-.374-7.846-1.119-2.241-10.077-24.288-13.809-33.256-3.635-8.733-7.327-7.55-10.077-7.688-2.609-.13-5.598-.158-8.583-.158-2.986.0-7.839 1.121-11.944 5.604-4.105 4.484-15.675 15.32-15.675 37.364.0 22.046 16.048 43.342 18.287 46.332 2.24 2.99 31.582 48.227 76.511 67.627 10.685 4.615 19.028 7.371 25.533 9.434 10.728 3.41 20.492 2.929 28.209 1.775 8.605-1.285 26.499-10.833 30.231-21.295 3.732-10.464 3.732-19.431 2.612-21.298-1.119-1.869-4.105-2.99-8.583-5.232z"/></svg></a><a target=_blank rel="noopener noreferrer" aria-label="share Tips GO on telegram" href="https://telegram.me/share/url?text=Tips%20GO&url=https%3a%2f%2fxavierzheng.github.io%2fposts%2ftips-go%2f"><svg viewBox="2 2 28 28" height="30" width="30" fill="currentcolor"><path d="M26.49 29.86H5.5a3.37 3.37.0 01-2.47-1 3.35 3.35.0 01-1-2.47V5.48A3.36 3.36.0 013 3 3.37 3.37.0 015.5 2h21A3.38 3.38.0 0129 3a3.36 3.36.0 011 2.46V26.37a3.35 3.35.0 01-1 2.47 3.38 3.38.0 01-2.51 1.02zm-5.38-6.71a.79.79.0 00.85-.66L24.73 9.24a.55.55.0 00-.18-.46.62.62.0 00-.41-.17q-.08.0-16.53 6.11a.59.59.0 00-.41.59.57.57.0 00.43.52l4 1.24 1.61 4.83a.62.62.0 00.63.43.56.56.0 00.4-.17L16.54 20l4.09 3A.9.9.0 0021.11 23.15zM13.8 20.71l-1.21-4q8.72-5.55 8.78-5.55c.15.0.23.0.23.16a.18.18.0 010 .06s-2.51 2.3-7.52 6.8z"/></svg></a></div></footer></article></main><footer class=footer><span>&copy; 2022 <a href=https://xavierzheng.github.io>Biology and computer science</a></span>
<span>Powered by
<a href=https://gohugo.io/ rel="noopener noreferrer" target=_blank>Hugo</a> &
        <a href=https://github.com/adityatelange/hugo-PaperMod/ rel=noopener target=_blank>PaperMod</a></span></footer><a href=#top aria-label="go to top" title="Go to Top (Alt + G)" class=top-link id=top-link accesskey=g><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 6" fill="currentcolor"><path d="M12 6H0l6-6z"/></svg></a><script>let menu=document.getElementById("menu");menu&&(menu.scrollLeft=localStorage.getItem("menu-scroll-position"),menu.onscroll=function(){localStorage.setItem("menu-scroll-position",menu.scrollLeft)}),document.querySelectorAll('a[href^="#"]').forEach(e=>{e.addEventListener("click",function(e){e.preventDefault();var t=this.getAttribute("href").substr(1);window.matchMedia("(prefers-reduced-motion: reduce)").matches?document.querySelector(`[id='${decodeURIComponent(t)}']`).scrollIntoView():document.querySelector(`[id='${decodeURIComponent(t)}']`).scrollIntoView({behavior:"smooth"}),t==="top"?history.replaceState(null,null," "):history.pushState(null,null,`#${t}`)})})</script><script>var mybutton=document.getElementById("top-link");window.onscroll=function(){document.body.scrollTop>800||document.documentElement.scrollTop>800?(mybutton.style.visibility="visible",mybutton.style.opacity="1"):(mybutton.style.visibility="hidden",mybutton.style.opacity="0")}</script><script>document.getElementById("theme-toggle").addEventListener("click",()=>{document.body.className.includes("dark")?(document.body.classList.remove("dark"),localStorage.setItem("pref-theme","light")):(document.body.classList.add("dark"),localStorage.setItem("pref-theme","dark"))})</script><script>document.querySelectorAll("pre > code").forEach(e=>{const n=e.parentNode.parentNode,t=document.createElement("button");t.classList.add("copy-code"),t.innerHTML="copy";function s(){t.innerHTML="copied!",setTimeout(()=>{t.innerHTML="copy"},2e3)}t.addEventListener("click",t=>{if("clipboard"in navigator){navigator.clipboard.writeText(e.textContent),s();return}const n=document.createRange();n.selectNodeContents(e);const o=window.getSelection();o.removeAllRanges(),o.addRange(n);try{document.execCommand("copy"),s()}catch{}o.removeRange(n)}),n.classList.contains("highlight")?n.appendChild(t):n.parentNode.firstChild==n||(e.parentNode.parentNode.parentNode.parentNode.parentNode.nodeName=="TABLE"?e.parentNode.parentNode.parentNode.parentNode.parentNode.appendChild(t):e.parentNode.appendChild(t))})</script></body></html>