<!DOCTYPE html>
<html lang="en" dir="auto">

<head><meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="robots" content="index, follow">
<title>Tips GO | Biology and computer science</title>
<meta name="keywords" content="GO">
<meta name="description" content="GO注意事項 Ref: Nine quick tips for pathway enrichment analysis, 2022 PLOS computation biology, https://doi.org/10.1371/journal.pcbi.1010348
Tip 1: Before starting, clarify which analysis you would like to perform What analysis type Pathway enrichment analysis (PEA) does not give clues about the active or inhibited status of the pathways. More appropriately, PEA provides information about how genes help carry out pathways.
overrepresentation analysis (ORA) focus of ORA methods is the gene set
ORA outputs all pathways enriched in the query gene list as a whole, and mainly uses a nonranked list">
<meta name="author" content="Me">
<link rel="canonical" href="https://canonical.url/to/page">
<link crossorigin="anonymous" href="/assets/css/stylesheet.bccfefac377bc340f06c260aed1bddf49a4354816d7c570d6aac75a997986c95.css" integrity="sha256-vM/vrDd7w0DwbCYK7Rvd9JpDVIFtfFcNaqx1qZeYbJU=" rel="preload stylesheet" as="style">
<script defer crossorigin="anonymous" src="/assets/js/highlight.f413e19d0714851f6474e7ee9632408e58ac146fbdbe62747134bea2fa3415e0.js" integrity="sha256-9BPhnQcUhR9kdOfuljJAjlisFG&#43;9vmJ0cTS&#43;ovo0FeA="
    onload="hljs.initHighlightingOnLoad();"></script>
<link rel="icon" href="https://xavierzheng.github.io/favicon.ico">
<link rel="icon" type="image/png" sizes="16x16" href="https://xavierzheng.github.io/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="https://xavierzheng.github.io/favicon-32x32.png">
<link rel="apple-touch-icon" href="https://xavierzheng.github.io/apple-touch-icon.png">
<link rel="mask-icon" href="https://xavierzheng.github.io/safari-pinned-tab.svg">
<meta name="theme-color" content="#2e2e33">
<meta name="msapplication-TileColor" content="#2e2e33">
<noscript>
    <style>
        #theme-toggle,
        .top-link {
            display: none;
        }

    </style>
    <style>
        @media (prefers-color-scheme: dark) {
            :root {
                --theme: rgb(29, 30, 32);
                --entry: rgb(46, 46, 51);
                --primary: rgb(218, 218, 219);
                --secondary: rgb(155, 156, 157);
                --tertiary: rgb(65, 66, 68);
                --content: rgb(196, 196, 197);
                --hljs-bg: rgb(46, 46, 51);
                --code-bg: rgb(55, 56, 62);
                --border: rgb(51, 51, 51);
            }

            .list {
                background: var(--theme);
            }

            .list:not(.dark)::-webkit-scrollbar-track {
                background: 0 0;
            }

            .list:not(.dark)::-webkit-scrollbar-thumb {
                border-color: var(--theme);
            }
        }

    </style>
</noscript><meta property="og:title" content="Tips GO" />
<meta property="og:description" content="GO注意事項 Ref: Nine quick tips for pathway enrichment analysis, 2022 PLOS computation biology, https://doi.org/10.1371/journal.pcbi.1010348
Tip 1: Before starting, clarify which analysis you would like to perform What analysis type Pathway enrichment analysis (PEA) does not give clues about the active or inhibited status of the pathways. More appropriately, PEA provides information about how genes help carry out pathways.
overrepresentation analysis (ORA) focus of ORA methods is the gene set
ORA outputs all pathways enriched in the query gene list as a whole, and mainly uses a nonranked list" />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://xavierzheng.github.io/posts/tips-go/" />
<meta property="og:image" content="https://xavierzheng.github.io/%3Cimage%20path/url%3E" /><meta property="article:section" content="posts" />
<meta property="article:published_time" content="2022-12-28T11:51:48+08:00" />
<meta property="article:modified_time" content="2022-12-28T11:51:48+08:00" />

<meta name="twitter:card" content="summary_large_image" />
<meta name="twitter:image" content="https://xavierzheng.github.io/%3Cimage%20path/url%3E" />
<meta name="twitter:title" content="Tips GO"/>
<meta name="twitter:description" content="GO注意事項 Ref: Nine quick tips for pathway enrichment analysis, 2022 PLOS computation biology, https://doi.org/10.1371/journal.pcbi.1010348
Tip 1: Before starting, clarify which analysis you would like to perform What analysis type Pathway enrichment analysis (PEA) does not give clues about the active or inhibited status of the pathways. More appropriately, PEA provides information about how genes help carry out pathways.
overrepresentation analysis (ORA) focus of ORA methods is the gene set
ORA outputs all pathways enriched in the query gene list as a whole, and mainly uses a nonranked list"/>


<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [, 
    {
      "@type": "ListItem",
      "position":  2 ,
      "name": "Posts",
      "item": "https://xavierzheng.github.io/posts/"
    }, 
    {
      "@type": "ListItem",
      "position":  3 ,
      "name": "Tips GO",
      "item": "https://xavierzheng.github.io/posts/tips-go/"
    }
  ]
}
</script>
<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "headline": "Tips GO",
  "name": "Tips GO",
  "description": "GO注意事項 Ref: Nine quick tips for pathway enrichment analysis, 2022 PLOS computation biology, https://doi.org/10.1371/journal.pcbi.1010348\nTip 1: Before starting, clarify which analysis you would like to perform What analysis type Pathway enrichment analysis (PEA) does not give clues about the active or inhibited status of the pathways. More appropriately, PEA provides information about how genes help carry out pathways.\noverrepresentation analysis (ORA) focus of ORA methods is the gene set\nORA outputs all pathways enriched in the query gene list as a whole, and mainly uses a nonranked list",
  "keywords": [
    "GO"
  ],
  "articleBody": "GO注意事項 Ref: Nine quick tips for pathway enrichment analysis, 2022 PLOS computation biology, https://doi.org/10.1371/journal.pcbi.1010348\nTip 1: Before starting, clarify which analysis you would like to perform What analysis type Pathway enrichment analysis (PEA) does not give clues about the active or inhibited status of the pathways. More appropriately, PEA provides information about how genes help carry out pathways.\noverrepresentation analysis (ORA) focus of ORA methods is the gene set\nORA outputs all pathways enriched in the query gene list as a whole, and mainly uses a nonranked list\n(except one option in g:Profiler g:GOst using a minimum hypergeometric value-based method).\ngene set enrichment analysis (GSEA) approaches 代表某一個pathway在transcriptome資料裡面有enrich focus of GSEA techniques is the ranked pathways list it is advisable to choose GSEA methods when there is uncertainty about the cutoff value. GSEA indicates the pathways that are enriched in genes located at both extreme ends of a ranked gene list, and a higher ranked pathway indicates that more genes are located at the very top or at the very end of this list competitive methods: BioPAX-Parser (BiP) [35] pathDIP [36,37] SPIA [38] CePaORA [39] PathNet [40] self-contained methods: CePa [39] GSEA [25–27] Which data type For unordered lists of genes, researchers can use g:Profiler g:GOSt [8–10], Enrichr [28,29], and BioPAX-Parser [35,47]. If the genes are ranked, g:Profiler g:GOSt can treasure this information and generate rank-based functional enrichment results. **If the input data are gene expression levels, they can be analyzed through GSEA [27]. ** 如果是expression level, 可以直接用GSEA pathDIP [37], instead, can assist with curated analyses based on scientific literature. Tip 2: Ensure the quality of your input genes or genomic regions 請確定放進去的是對的ID，有意義的list，有意義的expression\nTip 3: Use multiple PEA tools, not only one we suggest all the practitioners performing a functional enrichment analysis phase to employ at least two different PEA tools\nif the user had an unranked list of gene symbols, we would suggest to apply g:Profiler g:GOSt [10], Enrichr [28], and GeneTrail [52] to it, and then compare their results. To perform pathway enrichment by employing more than a single database, users can employ cPEA [48], a software tool able to deal with several pathway databases using the BioPAX language [61] to store and represent pathways. Or they can use BiP [47] by selecting the “Whole PathwayCommon Data” option that will perform cross enrichment using the whole collection of automatically downloaded locally Pathway Common databases [62] coded in BioPAX. Tip 4: Document all your PEA tests and their details 記得要寫好筆記，所有使用的參數以及資料庫的版本都要清楚寫下來，這是dry lab的實驗紀錄\nBox 1: Example of PEA test details My test ID: 2022-02-04, h10:02 EST. My input genes: AK4, ALDOC, EGLN1, FAM162A, MTFP1, PDK1, PGK1. My input genes’ type: gene symbols. Source: D. Cangelosi and colleagues [69]. Disease: neuroblastoma. Tool: g:Profiler g:GOSt. Access: online via Google Chrome browser. Version: e104_eg51_p15_3922dba. URL: https://biit.cs.ut.ee/gprofiler/gost Organism: Homo sapiens. Query: unordered genes. Statistical domain scope: only annotated genes. Significance threshold: g:SCS threshold. User threshold: 0.005. Data sources: default. All the other parameters: default. My output file(s) name(s): gprofiler_gost_NB_2022-02-04_h1002_output.csv My output file(s) folder: /home/davide/PEA_analyses/neuroblastoma/ My output file(s) location: bioinformatics-laptop-2021 (Dell Latitude E5420). Tip 5: Always use the corrected p-value, and not the nominal one terms adjusted p-values, corrected p-values, and false discovery rate (FDR) values are often used as synonyms in the scientific literature we suggest using the adjusted p-value threshold at 0.005 (that corresponds to 5 × 10−3), as recommended by Benjamin and colleagues [75]. We therefore suggest using the p.adj \u003c 0.005 threshold for a first strict analysis of the results, and then repeating the test by using a more permissive threshold such as p.adj \u003c 0.01, and then again with an even higher threshold, such as the traditional p.adj \u003c 0.05. Based on the characteristics of the experiment, results found by one particular threshold might be more suitable than results found with other thresholds. 記得從嚴格到寬鬆的原則 Tip 6: Keep in mind that your PEA results can be strongly affected by the statistical tests and the visualization techniques you use Statistical tests. 統計方法不一樣，會得到不一樣的結果 沒有最適合的方法，只有適合你實驗設計的方法 Visualization Enrichment Maps [80] and enrichplot [81] for biological pathways, AutoAnnotate [82] for networks, REVIGO [83] and CirGO [84] for GO annotations are few examples of different visualization techniques and contents. Network visualization techniques can also be used to detect a lower adjusted p-value threshold (Tip 5). Tip 7: Consider using subgroups of correlated genes instead of all your input genes Instead of using all genes as input for the PEA, we therefore suggest bioinformatics practitioners to detect subgroups of correlated genes and perform the PEA on each of these subgroups alone.\n不要一口氣全部做，可以考慮subgroup\nSubgroups of correlated genes can be found, for example, through protein–protein interaction networks’ tools such as IID [85–88], STRING [89–92], GeneMANIA [93–97], or Reactome Functional Interaction Network (Reactome FI) [98,99]. some R packages have been recently released: pathfindR [102] and netGO [103], which exploit the protein–protein interaction networks to produce more accurate PEA results. Tip 8: Use the (recent) scientific literature to review your PEA results we therefore suggest any practitioner to manually perform a literature search and look for scientific studies published about the significant genes–pathways associations found by the functional enrichment analysis and about the role of the genes inside the enriched pathways found.\n請找paper確定文獻是否更新\nTip 9: Ask a wet lab biologist or a clinician to review your PEA results a wet lab biologist or a clinician should review these results and clearly say if they make sense or if they contain mistakes or inappropriate information.\n請找專家用眼睛看\n",
  "wordCount" : "901",
  "inLanguage": "en",
  "image":"https://xavierzheng.github.io/%3Cimage%20path/url%3E","datePublished": "2022-12-28T11:51:48+08:00",
  "dateModified": "2022-12-28T11:51:48+08:00",
  "author":{
    "@type": "Person",
    "name": "Me"
  },
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "https://xavierzheng.github.io/posts/tips-go/"
  },
  "publisher": {
    "@type": "Organization",
    "name": "Biology and computer science",
    "logo": {
      "@type": "ImageObject",
      "url": "https://xavierzheng.github.io/favicon.ico"
    }
  }
}
</script>
</head>

<body class="" id="top">
<script>
    if (localStorage.getItem("pref-theme") === "dark") {
        document.body.classList.add('dark');
    } else if (localStorage.getItem("pref-theme") === "light") {
        document.body.classList.remove('dark')
    } else if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
        document.body.classList.add('dark');
    }

</script>

<header class="header">
    <nav class="nav">
        <div class="logo">
            <a href="https://xavierzheng.github.io" accesskey="h" title="Biology and computer science (Alt + H)">Biology and computer science</a>
            <div class="logo-switches">
                <button id="theme-toggle" accesskey="t" title="(Alt + T)">
                    <svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
                    </svg>
                    <svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <circle cx="12" cy="12" r="5"></circle>
                        <line x1="12" y1="1" x2="12" y2="3"></line>
                        <line x1="12" y1="21" x2="12" y2="23"></line>
                        <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
                        <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
                        <line x1="1" y1="12" x2="3" y2="12"></line>
                        <line x1="21" y1="12" x2="23" y2="12"></line>
                        <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
                        <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
                    </svg>
                </button>
            </div>
        </div>
        <ul id="menu">
        </ul>
    </nav>
</header>
<main class="main">

<article class="post-single">
  <header class="post-header">
    <div class="breadcrumbs"><a href="https://xavierzheng.github.io">Home</a>&nbsp;»&nbsp;<a href="https://xavierzheng.github.io/posts/">Posts</a></div>
    <h1 class="post-title">
      Tips GO
    </h1>
    <div class="post-meta"><span title='2022-12-28 11:51:48 +0800 CST'>December 28, 2022</span>&nbsp;·&nbsp;Me

</div>
  </header> <div class="toc">
    <details  open>
        <summary accesskey="c" title="(Alt + C)">
            <span class="details">Table of Contents</span>
        </summary>

        <div class="inner"><nav id="TableOfContents">
  <ul>
    <li><a href="#tip-1-before-starting-clarify-which-analysis-you-would-like-to-perform">Tip 1: Before starting, clarify which analysis you would like to perform</a>
      <ul>
        <li><a href="#what-analysis-type">What analysis type</a></li>
        <li><a href="#which-data-type">Which data type</a></li>
      </ul>
    </li>
    <li><a href="#tip-2-ensure-the-quality-of-your-input-genes-or-genomic-regions">Tip 2: Ensure the quality of your input genes or genomic regions</a></li>
    <li><a href="#tip-3-use-multiple-pea-tools-not-only-one">Tip 3: Use multiple PEA tools, not only one</a></li>
    <li><a href="#tip-4-document-all-your-pea-tests-and-their-details">Tip 4: Document all your PEA tests and their details</a></li>
    <li><a href="#tip-5-always-use-the-corrected-p-value-and-not-the-nominal-one">Tip 5: Always use the corrected <em>p</em>-value, and not the nominal one</a></li>
    <li><a href="#tip-6-keep-in-mind-that-your-pea-results-can-be-strongly-affected--by-the-statistical-tests-and-the-visualization-techniques-you-use">Tip 6: Keep in mind that your PEA results can be strongly affected  by the statistical tests and the visualization techniques you use</a>
      <ul>
        <li><a href="#statistical-tests">Statistical tests.</a></li>
        <li><a href="#visualization">Visualization</a></li>
      </ul>
    </li>
    <li><a href="#tip-7-consider-using-subgroups-of-correlated-genes-instead-of-all-your-input-genes">Tip 7: Consider using subgroups of correlated genes instead of all your input genes</a></li>
    <li><a href="#tip-8-use-the-recent-scientific-literature-to-review-your-pea-results">Tip 8: Use the (recent) scientific literature to review your PEA results</a></li>
    <li><a href="#tip-9-ask-a-wet-lab-biologist-or-a-clinician-to-review-your-pea-results">Tip 9: Ask a wet lab biologist or a clinician to review your PEA results</a></li>
  </ul>
</nav>
        </div>
    </details>
</div>

  <div class="post-content"><h1 id="go注意事項">GO注意事項<a hidden class="anchor" aria-hidden="true" href="#go注意事項">#</a></h1>
<p><strong>Ref</strong>: Nine quick tips for pathway enrichment analysis, 2022 PLOS computation biology, <a href="https://doi.org/10.1371/journal.pcbi.1010348">https://doi.org/10.1371/journal.pcbi.1010348</a></p>
<h2 id="tip-1-before-starting-clarify-which-analysis-you-would-like-to-perform">Tip 1: Before starting, clarify which analysis you would like to perform<a hidden class="anchor" aria-hidden="true" href="#tip-1-before-starting-clarify-which-analysis-you-would-like-to-perform">#</a></h2>
<h3 id="what-analysis-type">What analysis type<a hidden class="anchor" aria-hidden="true" href="#what-analysis-type">#</a></h3>
<p>Pathway enrichment analysis (PEA) does not give clues about the active or inhibited status of the  pathways. More appropriately, PEA provides information about how genes  help carry out pathways.</p>
<h4 id="overrepresentation-analysis-ora">overrepresentation analysis (ORA)<a hidden class="anchor" aria-hidden="true" href="#overrepresentation-analysis-ora">#</a></h4>
<ul>
<li>
<p><strong>focus of ORA methods is the gene set</strong></p>
</li>
<li>
<p>ORA outputs all pathways enriched in the query gene list as a whole,  and mainly uses a nonranked list</p>
</li>
<li>
<p>(except one option in g:Profiler g:GOst using a minimum hypergeometric value-based method).</p>
</li>
</ul>
<h4 id="gene-set-enrichment-analysis-gsea-approaches">gene set enrichment analysis (GSEA) approaches<a hidden class="anchor" aria-hidden="true" href="#gene-set-enrichment-analysis-gsea-approaches">#</a></h4>
<ul>
<li>代表某一個pathway在transcriptome資料裡面有enrich</li>
<li><strong>focus of GSEA techniques is the ranked pathways list</strong></li>
<li>it is advisable to choose GSEA methods when there is uncertainty about  the cutoff value.</li>
<li>GSEA indicates the pathways that are enriched in genes located at both  extreme ends of a ranked gene list, and a higher ranked pathway  indicates that more genes are located at the very top or at the very end of this list
<ul>
<li>competitive methods:</li>
<li>BioPAX-Parser (BiP) [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref035">35</a>]</li>
<li>pathDIP [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref036">36</a>,<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref037">37</a>]</li>
<li>SPIA [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref038">38</a>]</li>
<li>CePaORA [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref039">39</a>]</li>
<li>PathNet [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref040">40</a>]</li>
<li>self-contained methods:</li>
<li>CePa [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref039">39</a>]</li>
<li>GSEA [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref025">25</a>–<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref027">27</a>]</li>
</ul>
</li>
</ul>
<h3 id="which-data-type">Which data type<a hidden class="anchor" aria-hidden="true" href="#which-data-type">#</a></h3>
<ul>
<li>For unordered lists of genes, researchers can use g:Profiler g:GOSt [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref008">8</a>–<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref010">10</a>], Enrichr [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref028">28</a>,<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref029">29</a>], and BioPAX-Parser [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref035">35</a>,<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref047">47</a>].</li>
<li>If the genes are ranked, g:Profiler g:GOSt can treasure this  information and generate rank-based functional enrichment results.</li>
<li>**If  the input data are gene expression levels, they can be analyzed through  GSEA [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref027">27</a>]. **</li>
<li><strong>如果是expression level, 可以直接用GSEA</strong></li>
<li>pathDIP [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref037">37</a>], instead, can assist with curated analyses based on scientific literature.</li>
</ul>
<h2 id="tip-2-ensure-the-quality-of-your-input-genes-or-genomic-regions">Tip 2: Ensure the quality of your input genes or genomic regions<a hidden class="anchor" aria-hidden="true" href="#tip-2-ensure-the-quality-of-your-input-genes-or-genomic-regions">#</a></h2>
<p>請確定放進去的是對的ID，有意義的list，有意義的expression</p>
<h2 id="tip-3-use-multiple-pea-tools-not-only-one">Tip 3: Use multiple PEA tools, not only one<a hidden class="anchor" aria-hidden="true" href="#tip-3-use-multiple-pea-tools-not-only-one">#</a></h2>
<p>we suggest all the practitioners performing a functional enrichment analysis phase to employ at least two different PEA tools</p>
<ul>
<li>if the user had an unranked list of gene symbols, we would suggest to apply g:Profiler g:GOSt [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref010">10</a>], Enrichr [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref028">28</a>], and GeneTrail [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref052">52</a>] to it, and then compare their results.</li>
<li>To perform pathway enrichment by employing more than a single database, users can employ cPEA [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref048">48</a>], a software tool able to deal with several pathway databases using the BioPAX language [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref061">61</a>] to store and represent pathways.</li>
<li>Or they can use BiP [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref047">47</a>] by selecting the “Whole PathwayCommon Data” option that will perform  cross enrichment using the whole collection of automatically downloaded  locally Pathway Common databases [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref062">62</a>] coded in BioPAX.</li>
</ul>
<h2 id="tip-4-document-all-your-pea-tests-and-their-details">Tip 4: Document all your PEA tests and their details<a hidden class="anchor" aria-hidden="true" href="#tip-4-document-all-your-pea-tests-and-their-details">#</a></h2>
<p>記得要寫好筆記，所有使用的參數以及資料庫的版本都要清楚寫下來，這是dry lab的實驗紀錄</p>
<pre tabindex="0"><code>Box 1: Example of PEA test details

My test ID: 2022-02-04, h10:02 EST.

My input genes: AK4, ALDOC, EGLN1, FAM162A, MTFP1, PDK1, PGK1.

My input genes’ type: gene symbols.

Source: D. Cangelosi and colleagues [69].

Disease: neuroblastoma.

Tool: g:Profiler g:GOSt.

Access: online via Google Chrome browser.

Version: e104_eg51_p15_3922dba.

URL: https://biit.cs.ut.ee/gprofiler/gost

Organism: Homo sapiens.

Query: unordered genes.

Statistical domain scope: only annotated genes.

Significance threshold: g:SCS threshold.

User threshold: 0.005.

Data sources: default.

All the other parameters: default.

My output file(s) name(s): gprofiler_gost_NB_2022-02-04_h1002_output.csv

My output file(s) folder: /home/davide/PEA_analyses/neuroblastoma/

My output file(s) location: bioinformatics-laptop-2021 (Dell Latitude E5420).
</code></pre><h2 id="tip-5-always-use-the-corrected-p-value-and-not-the-nominal-one">Tip 5: Always use the corrected <em>p</em>-value, and not the nominal one<a hidden class="anchor" aria-hidden="true" href="#tip-5-always-use-the-corrected-p-value-and-not-the-nominal-one">#</a></h2>
<ul>
<li>terms adjusted p-values, corrected p-values, and false discovery rate  (FDR) values are often used as synonyms in the scientific literature</li>
<li>we suggest using the adjusted <em>p</em>-value threshold at 0.005 (that corresponds to 5 × 10−3), as recommended by Benjamin and colleagues [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref075">75</a>].</li>
<li>We therefore suggest using the <em>p</em>.<em>adj</em> &lt; 0.005  threshold for a first strict analysis of the results, and then repeating the test by using a more permissive threshold such as <em>p</em>.<em>adj</em> &lt; 0.01, and then again with an even higher threshold, such as the traditional <em>p</em>.<em>adj</em> &lt; 0.05. Based on the characteristics of the experiment, results  found by one particular threshold might be more suitable than results  found with other thresholds.</li>
<li>記得從嚴格到寬鬆的原則</li>
</ul>
<h2 id="tip-6-keep-in-mind-that-your-pea-results-can-be-strongly-affected--by-the-statistical-tests-and-the-visualization-techniques-you-use">Tip 6: Keep in mind that your PEA results can be strongly affected  by the statistical tests and the visualization techniques you use<a hidden class="anchor" aria-hidden="true" href="#tip-6-keep-in-mind-that-your-pea-results-can-be-strongly-affected--by-the-statistical-tests-and-the-visualization-techniques-you-use">#</a></h2>
<h3 id="statistical-tests">Statistical tests.<a hidden class="anchor" aria-hidden="true" href="#statistical-tests">#</a></h3>
<ul>
<li>統計方法不一樣，會得到不一樣的結果</li>
<li>沒有最適合的方法，只有適合你實驗設計的方法</li>
</ul>
<h3 id="visualization">Visualization<a hidden class="anchor" aria-hidden="true" href="#visualization">#</a></h3>
<ul>
<li>Enrichment Maps [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref080">80</a>] and enrichplot [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref081">81</a>] for biological pathways,</li>
<li>AutoAnnotate [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref082">82</a>] for networks,</li>
<li>REVIGO [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref083">83</a>] and CirGO [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref084">84</a>] for GO annotations are few examples of different visualization  techniques and contents.</li>
<li>Network visualization techniques can also be  used to detect a lower adjusted <em>p</em>-value threshold (Tip 5).</li>
</ul>
<h2 id="tip-7-consider-using-subgroups-of-correlated-genes-instead-of-all-your-input-genes">Tip 7: Consider using subgroups of correlated genes instead of all your input genes<a hidden class="anchor" aria-hidden="true" href="#tip-7-consider-using-subgroups-of-correlated-genes-instead-of-all-your-input-genes">#</a></h2>
<p>Instead of using all genes as input for the PEA, we therefore suggest  bioinformatics practitioners to detect subgroups of correlated genes and perform the PEA on each of these subgroups alone.</p>
<p>不要一口氣全部做，可以考慮subgroup</p>
<ul>
<li>Subgroups of correlated genes can be found, for example, through protein–protein interaction networks’ tools such as IID [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref085">85</a>–<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref088">88</a>], STRING [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref089">89</a>–<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref092">92</a>], GeneMANIA [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref093">93</a>–<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref097">97</a>], or Reactome Functional Interaction Network (Reactome FI) [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref098">98</a>,<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref099">99</a>].</li>
<li>some R packages have been recently released: <strong>pathfindR [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref102">102</a>] and netGO [<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348#pcbi.1010348.ref103">103</a>], which exploit the protein–protein interaction networks to produce more accurate PEA results.</strong></li>
</ul>
<h2 id="tip-8-use-the-recent-scientific-literature-to-review-your-pea-results">Tip 8: Use the (recent) scientific literature to review your PEA results<a hidden class="anchor" aria-hidden="true" href="#tip-8-use-the-recent-scientific-literature-to-review-your-pea-results">#</a></h2>
<p>we therefore suggest any practitioner to manually perform a literature  search and look for scientific studies published about the significant  genes–pathways associations found by the functional enrichment analysis  and about the role of the genes inside the enriched pathways found.</p>
<p>請找paper確定文獻是否更新</p>
<h2 id="tip-9-ask-a-wet-lab-biologist-or-a-clinician-to-review-your-pea-results">Tip 9: Ask a wet lab biologist or a clinician to review your PEA results<a hidden class="anchor" aria-hidden="true" href="#tip-9-ask-a-wet-lab-biologist-or-a-clinician-to-review-your-pea-results">#</a></h2>
<p>a wet lab biologist or a clinician should review these results and  clearly say if they make sense or if they contain mistakes or  inappropriate information.</p>
<p>請找專家用眼睛看</p>


  </div>

  <footer class="post-footer">
    <ul class="post-tags">
      <li><a href="https://xavierzheng.github.io/tags/go/">GO</a></li>
    </ul>

  </footer>
</article>
    </main>
    
<footer class="footer">
    <span>&copy; 2022 <a href="https://xavierzheng.github.io">Biology and computer science</a></span>
    <span>
        Powered by
        <a href="https://gohugo.io/" rel="noopener noreferrer" target="_blank">Hugo</a> &
        <a href="https://github.com/adityatelange/hugo-PaperMod/" rel="noopener" target="_blank">PaperMod</a>
    </span>
</footer>
<a href="#top" aria-label="go to top" title="Go to Top (Alt + G)" class="top-link" id="top-link" accesskey="g">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 6" fill="currentColor">
        <path d="M12 6H0l6-6z" />
    </svg>
</a>

<script>
    let menu = document.getElementById('menu')
    if (menu) {
        menu.scrollLeft = localStorage.getItem("menu-scroll-position");
        menu.onscroll = function () {
            localStorage.setItem("menu-scroll-position", menu.scrollLeft);
        }
    }

    document.querySelectorAll('a[href^="#"]').forEach(anchor => {
        anchor.addEventListener("click", function (e) {
            e.preventDefault();
            var id = this.getAttribute("href").substr(1);
            if (!window.matchMedia('(prefers-reduced-motion: reduce)').matches) {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView({
                    behavior: "smooth"
                });
            } else {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView();
            }
            if (id === "top") {
                history.replaceState(null, null, " ");
            } else {
                history.pushState(null, null, `#${id}`);
            }
        });
    });

</script>
<script>
    var mybutton = document.getElementById("top-link");
    window.onscroll = function () {
        if (document.body.scrollTop > 800 || document.documentElement.scrollTop > 800) {
            mybutton.style.visibility = "visible";
            mybutton.style.opacity = "1";
        } else {
            mybutton.style.visibility = "hidden";
            mybutton.style.opacity = "0";
        }
    };

</script>
<script>
    document.getElementById("theme-toggle").addEventListener("click", () => {
        if (document.body.className.includes("dark")) {
            document.body.classList.remove('dark');
            localStorage.setItem("pref-theme", 'light');
        } else {
            document.body.classList.add('dark');
            localStorage.setItem("pref-theme", 'dark');
        }
    })

</script>
</body>

</html>
